From cd627a1bf03ee89f19395b7840f7455e236417cf Mon Sep 17 00:00:00 2001
From: Tails developers <tails@boum.org>
Date: Thu, 16 Aug 2012 01:17:07 +0200
Subject: [PATCH 2/7] Only write a nice title

---
 init.c |    4 ++--
 lib.c  |    9 ++++++++-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/init.c b/init.c
index f7f43c1..62a30af 100644
--- a/init.c
+++ b/init.c
@@ -109,10 +109,10 @@ static void display_init(void)
 	}
 
 	/* Make the name background red */
-	for(i=0, pp=(char *)(SCREEN_ADR+1); i<TITLE_WIDTH; i++, pp+=2) {
+	for(i=1, pp=(char *)(SCREEN_ADR+1); i<38; i++, pp+=2) {
 		*pp = 0x20;
 	}
-	cprint(0, 0, "      Memtest86  v4.20      ");
+	real_cprint(0, 0, "      Wiping memory, please wait      ");
 
 	for(i=0, pp=(char *)(SCREEN_ADR+1); i<2; i++, pp+=30) {
 		*pp = 0xA4;
diff --git a/lib.c b/lib.c
index d107039..cc8993d 100644
--- a/lib.c
+++ b/lib.c
@@ -236,10 +236,17 @@ void clear_scroll(void)
 }
 
 /*
- * Print characters on screen
+ * Do nothing si the memtest inteface is not showed
  */
 void cprint(int y, int x, const char *text)
 {
+}
+
+/*
+ * Actually print characters on screen, this is the old cprint
+ */
+void real_cprint(int y, int x, const char *text)
+{
 	register int i;
 	char *dptr;
 
-- 
1.7.10.4

