From 9ac09b93bda5cfb05b33b6d528f11c2ce0a3e1a0 Mon Sep 17 00:00:00 2001
From: Tails developers <tails@boum.org>
Date: Thu, 16 Aug 2012 01:18:43 +0200
Subject: [PATCH 4/7] Actually writes the progress bar, and move it up

---
 error.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/error.c b/error.c
index 5b63be0..affdc85 100644
--- a/error.c
+++ b/error.c
@@ -455,7 +455,7 @@ void do_tick(void)
 		if (v->tptr >= BAR_SIZE) {
 			break;
 		}
-		cprint(1, COL_MID+9+v->tptr, "#");
+		real_cprint(0, COL_MID+9+v->tptr, "#");
 		v->tptr++;
 	}
 	
-- 
1.7.10.4

