diff -Naur syslinux/syslinux.orig/live_486.cfg syslinux/syslinux/live_486.cfg
--- syslinux/syslinux.orig/live_486.cfg	1970-01-01 01:00:00.000000000 +0100
+++ syslinux/syslinux/live_486.cfg	2012-09-01 11:48:36.000000000 +0200
@@ -0,0 +1,11 @@
+include menu.cfg
+
+label live
+	menu label Live
+	kernel /live/vmlinuz-486
+	append initrd=/live/initrd.img boot=live config live-media=removable nopersistent noprompt quiet timezone=Etc/UTC block.events_dfl_poll_msecs=1000 splash nox11autologin  quiet
+
+label livefailsafe
+	menu label Live (failsafe)
+	kernel /live/vmlinuz-486
+	append initrd=/live/initrd.img boot=live config live-media=removable nopersistent noprompt quiet timezone=Etc/UTC block.events_dfl_poll_msecs=1000 splash nox11autologin  noapic noapm nodma nomce nolapic nomodeset nosmp vga=normal
diff -Naur syslinux/syslinux.orig/live_686-pae.cfg syslinux/syslinux/live_686-pae.cfg
--- syslinux/syslinux.orig/live_686-pae.cfg	1970-01-01 01:00:00.000000000 +0100
+++ syslinux/syslinux/live_686-pae.cfg	2012-09-01 11:48:40.000000000 +0200
@@ -0,0 +1,11 @@
+include menu.cfg
+
+label live
+	menu label Live
+	kernel /live/vmlinuz-686-pae
+	append initrd=/live/initrd.img boot=live config live-media=removable nopersistent noprompt quiet timezone=Etc/UTC block.events_dfl_poll_msecs=1000 splash nox11autologin  quiet
+
+label livefailsafe
+	menu label Live (failsafe)
+	kernel /live/vmlinuz-686-pae
+	append initrd=/live/initrd.img boot=live config live-media=removable nopersistent noprompt quiet timezone=Etc/UTC block.events_dfl_poll_msecs=1000 splash nox11autologin  noapic noapm nodma nomce nolapic nomodeset nosmp vga=normal
diff -Naur syslinux/syslinux.orig/live_amd64.cfg syslinux/syslinux/live_amd64.cfg
--- syslinux/syslinux.orig/live_amd64.cfg	1970-01-01 01:00:00.000000000 +0100
+++ syslinux/syslinux/live_amd64.cfg	2012-09-01 11:48:46.000000000 +0200
@@ -0,0 +1,11 @@
+include menu.cfg
+
+label live
+	menu label Live
+	kernel /live/vmlinuz-amd64
+	append initrd=/live/initrd.img boot=live config live-media=removable nopersistent noprompt quiet timezone=Etc/UTC block.events_dfl_poll_msecs=1000 splash nox11autologin  quiet
+
+label livefailsafe
+	menu label Live (failsafe)
+	kernel /live/vmlinuz-amd64
+	append initrd=/live/initrd.img boot=live config live-media=removable nopersistent noprompt quiet timezone=Etc/UTC block.events_dfl_poll_msecs=1000 splash nox11autologin  noapic noapm nodma nomce nolapic nomodeset nosmp vga=normal
diff -Naur syslinux/syslinux.orig/menu.cfg syslinux/syslinux/menu.cfg
--- syslinux/syslinux.orig/menu.cfg	2012-09-01 11:23:58.000000000 +0200
+++ syslinux/syslinux/menu.cfg	2012-09-01 11:54:46.000000000 +0200
@@ -1,7 +1,9 @@
+prompt 0
+timeout 40
+
 menu hshift 15
 menu width 49
 
 menu title Boot Tails
 include stdmenu.cfg
 include tails.cfg
-include live.cfg
diff -Naur syslinux/syslinux.orig/syslinux.cfg syslinux/syslinux/syslinux.cfg
--- syslinux/syslinux.orig/syslinux.cfg	2012-09-01 11:23:58.000000000 +0200
+++ syslinux/syslinux/syslinux.cfg	2012-09-01 11:54:52.000000000 +0200
@@ -1,4 +1,14 @@
-include menu.cfg
-default /syslinux/vesamenu.c32
+label select_menu
+	com32 /syslinux/ifcpu64.c32
+	append menu_amd64 -- menu_686-pae -- menu_486
+label menu_amd64
+	kernel /syslinux/vesamenu.c32
+	append live_amd64.cfg
+label menu_686-pae
+	kernel /syslinux/vesamenu.c32
+	append live_686-pae.cfg
+label menu_486
+	kernel /syslinux/vesamenu.c32
+	append live_486.cfg
+default select_menu
 prompt 0
-timeout 40
