From 633d8f976fc2975e56d0d2aca243806794a17951 Mon Sep 17 00:00:00 2001
From: anonym <anonym@riseup.net>
Date: Wed, 27 Feb 2019 09:45:04 +0100
Subject: [PATCH 1/4] Prefer fetched configurations using SSL over plaintext.

---
 .../components/accountcreation/content/readFromXML.js  | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/comm/mail/components/accountcreation/content/readFromXML.js b/comm/mail/components/accountcreation/content/readFromXML.js
index 95019f97d45..d0fa70846c9 100644
--- a/comm/mail/components/accountcreation/content/readFromXML.js
+++ b/comm/mail/components/accountcreation/content/readFromXML.js
@@ -102,7 +102,10 @@ function readFromXML(clientConfigXML, subSource) {
             SSL: 2,
             STARTTLS: 3,
           });
-          break; // take first that we support
+          if (iO.socketType != 1) {
+            // pick first non-plaintext protocol, if available
+            break;
+          }
         } catch (e) {
           exception = e;
         }
@@ -235,7 +238,10 @@ function readFromXML(clientConfigXML, subSource) {
             SSL: 2,
             STARTTLS: 3,
           });
-          break; // take first that we support
+          if (oO.socketType != 1) {
+            // pick first non-plaintext protocol, if available
+            break;
+          }
         } catch (e) {
           exception = e;
         }
-- 
2.28.0

