--- a/usr/share/webext/ublock-origin/js/background.js	2018-08-10 06:18:58.000000000 +0000
+++ b/usr/share/webext/ublock-origin/js/background.js	2018-08-13 07:39:40.029408152 +0000
@@ -74,7 +74,7 @@
         userSettings: {
             advancedUserEnabled: false,
             alwaysDetachLogger: true,
-            autoUpdate: true,
+            autoUpdate: false,
             cloudStorageEnabled: false,
             collapseBlocked: true,
             colorBlindFriendly: false,

