--- a/usr/share/mozilla/extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/uBlock0@raymondhill.net/js/background.js	2019-07-07 21:57:21.000000000 +0000
+++ b/usr/share/mozilla/extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}/uBlock0@raymondhill.net/js/background.js	2019-07-08 10:21:49.658415914 +0000
@@ -69,7 +69,7 @@
         userSettings: {
             advancedUserEnabled: false,
             alwaysDetachLogger: true,
-            autoUpdate: true,
+            autoUpdate: false,
             cloudStorageEnabled: false,
             collapseBlocked: true,
             colorBlindFriendly: false,
