--- a/usr/share/webext/ublock-origin/js/background.js	2018-10-11 17:14:14.000000000 +0200
+++ b/usr/share/webext/ublock-origin/js/background.js	2018-10-12 12:07:07.951778615 +0200
@@ -77,7 +77,7 @@
         userSettings: {
             advancedUserEnabled: false,
             alwaysDetachLogger: true,
-            autoUpdate: true,
+            autoUpdate: false,
             cloudStorageEnabled: false,
             collapseBlocked: true,
             colorBlindFriendly: false,
