From b76837578102a1d9163693c116d1098653effcd6 Mon Sep 17 00:00:00 2001
From: Tails developers <tails@boum.org>
Date: Thu, 16 Aug 2012 01:21:43 +0200
Subject: [PATCH 3/7] Remove most colors

---
 init.c |   16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/init.c b/init.c
index 62a30af..88ac408 100644
--- a/init.c
+++ b/init.c
@@ -102,22 +102,8 @@ static void display_init(void)
 	serial_echo_print("\x1B[0m");
 	serial_echo_print("\x1B[37m\x1B[44m");
 
-	/* Clear screen & set background to blue */
-	for(i=0, pp=(char *)(SCREEN_ADR); i<80*24; i++) {
-		*pp++ = ' ';
-		*pp++ = 0x17;
-	}
-
-	/* Make the name background red */
-	for(i=1, pp=(char *)(SCREEN_ADR+1); i<38; i++, pp+=2) {
-		*pp = 0x20;
-	}
-	real_cprint(0, 0, "      Wiping memory, please wait      ");
+	real_cprint(0, 0, " Wiping memory, please wait...");
 
-	for(i=0, pp=(char *)(SCREEN_ADR+1); i<2; i++, pp+=30) {
-		*pp = 0xA4;
-	}
-	cprint(0, 15, "+");
 
 	/* Do reverse video for the bottom display line */
 	for(i=0, pp=(char *)(SCREEN_ADR+1+(24 * 160)); i<80; i++, pp+=2) {
-- 
1.7.10.4

