From a8ec6d5c1e94ad08d199fa0655f3ffa64038157f Mon Sep 17 00:00:00 2001
From: Tails developers <tails@boum.org>
Date: Thu, 16 Aug 2012 01:39:08 +0200
Subject: [PATCH 5/7] Display message at the end

---
 main.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main.c b/main.c
index 6be644b..19639e5 100644
--- a/main.c
+++ b/main.c
@@ -493,8 +493,8 @@ void do_test(void)
 		cprint(0, COL_MID+8,
 			"                                         ");
 		if (v->ecount == 0 && v->testsel < 0) {
-			cprint(LINE_MSG+5, 0,
-			"              *****Pass complete, no errors, press Esc to exit*****            ");
+			real_cprint(12, 0,
+			"             ****Memory wipe successful, attempting to shutdown****            ");
 			if(BEEP_END_NO_ERROR) {
 				beep(1000);
 				beep(2000);
-- 
1.7.10.4

