--- a/etc/apparmor.d/abstractions/python
+++ b/etc/apparmor.d/abstractions/python
@@ -17,7 +17,7 @@
 
   /usr/local/lib{,32,64}/python{2.[4-7],3.[0-5]}/**.{pyc,so}           mr,
   /usr/local/lib{,32,64}/python{2.[4-7],3.[0-5]}/**.{egg,py,pth}       r,
-  /usr/local/lib{,32,64}/python{2.[4-7],3.[0-5]}/{site,dist}-packages/ r,
+  /usr/local/lib{,32,64}/python{2.[4-7],3,3.[0-5]}/{site,dist}-packages/ r,
   /usr/local/lib{,32,64}/python3.[0-5]/lib-dynload/*.so            mr,
 
   # Site-wide configuration
