--- a/etc/apparmor.d/usr.bin.vidalia	2015-06-10 09:15:34.668000000 +0000
+++ b/etc/apparmor.d/usr.bin.vidalia	2015-06-10 09:38:17.812000000 +0000
@@ -9,6 +9,8 @@
 
   owner @{HOME}/.vidalia/ rw,
   owner @{HOME}/.vidalia/** rwmk,
+  owner /lib/live/mount/rootfs/*.squashfs/home/vidalia/.vidalia/ rw,
+  owner /lib/live/mount/rootfs/*.squashfs/home/vidalia/.vidalia/** rwmk,
 
   /{var/,} r,
   /{var/,}run/ r,
@@ -22,6 +24,13 @@
   owner @{PROC}/[0-9]*/cmdline r,
   owner @{PROC}/[0-9]*/fd/ r,
 
+  deny /var/cache/fontconfig/ w,
+  /home/vidalia/.fontconfig/ rw,
+  /home/vidalia/.fontconfig/* rw,
+  /home/vidalia/.config/Trolltech.conf* rw,
+  /home/vidalia/.wh..wh..vidalia.*/ rw,
+  /lib/live/mount/overlay/home/vidalia/.wh..wh..vidalia.*/ rw,
+
   # Site-specific additions and overrides. See local/README for details.
   #include <local/usr.bin.vidalia>
 }

