--- /etc/apparmor.d/abstractions/freedesktop.org	2018-10-07 19:38:49.308000000 +0000
+++ /etc/apparmor.d/abstractions/freedesktop.org	2018-10-07 19:48:40.400000000 +0000
@@ -24,7 +24,7 @@
   /usr/local/share/pixmaps/**           r,
 
   # this should probably go elsewhere
-  /usr/share/mime/**              r,
+  /usr/{local/,}share/mime/**              r,
 
   # per-user configurations
   owner @{HOME}/.icons/                 r,
