diff -Naur etc/apparmor.d.orig/abstractions/base etc/apparmor.d/abstractions/base
--- a/etc/apparmor.d/abstractions/base	2018-02-25 18:23:21.000000000 +0000
+++ b/etc/apparmor.d/abstractions/base	2018-03-10 08:03:52.281671815 +0000
@@ -26,6 +26,7 @@
   /etc/locale/**                 r,
   /etc/locale.alias              r,
   /etc/localtime                 r,
+  /etc/tor/torsocks.conf         r,
   /etc/writable/localtime        r,
   /usr/share/locale-bundle/**    r,
   /usr/share/locale-langpack/**  r,
diff -Naur etc/apparmor.d.orig/abstractions/ubuntu-helpers etc/apparmor.d/abstractions/ubuntu-helpers
--- a/etc/apparmor.d/abstractions/ubuntu-helpers	2018-01-14 17:01:17.000000000 +0000
+++ b/etc/apparmor.d/abstractions/ubuntu-helpers	2018-08-26 17:37:23.000000000 +0000
@@ -67,8 +67,8 @@
   # in limited libraries so glibc's secure execution should be enough to not
   # require the santized_helper (ie, LD_PRELOAD will only use standard system
   # paths (man ld.so)).
-  /usr/lib/chromium-browser/chromium-browser-sandbox PUxr,
-  /usr/lib/chromium{,-browser}/chrome-sandbox PUxr,
+  # /usr/lib/chromium-browser/chromium-browser-sandbox PUxr,
+  # /usr/lib/chromium{,-browser}/chrome-sandbox PUxr,
   /opt/google/chrome{,-beta,-unstable}/chrome-sandbox PUxr,
   /opt/google/chrome{,-beta,-unstable}/google-chrome Pixr,
   /opt/google/chrome{,-beta,-unstable}/chrome Pixr,
