--- a/etc/apparmor.d/usr.bin.pidgin	2015-06-04 12:37:02.453412928 +0000
+++ b/etc/apparmor.d/usr.bin.pidgin	2015-06-04 12:37:40.309205204 +0000
@@ -13,7 +13,7 @@
   #include <abstractions/gnome>
   #include <abstractions/gstreamer>
   #include <abstractions/ibus>
-  #include <abstractions/launchpad-integration>
+  # #include <abstractions/launchpad-integration>
   #include <abstractions/nameservice>
   #include <abstractions/private-files-strict>
   #include <abstractions/ssl_certs>
@@ -68,6 +68,7 @@
   /usr/bin/gvfs-open rmix,
   /usr/bin/pidgin r,
   /usr/bin/xdg-open rmix,
+  /usr/local/bin/tor-browser rmUx,
 
   /etc/purple/prefs.xml r,
 
