--- a/etc/apparmor.d/usr.bin.pidgin	2015-06-04 12:37:02.453412928 +0000
+++ b/etc/apparmor.d/usr.bin.pidgin	2015-06-04 12:37:40.309205204 +0000
@@ -11,7 +11,7 @@
   #include <abstractions/enchant>
   #include <abstractions/gnome>
   #include <abstractions/ibus>
-  #include <abstractions/launchpad-integration>
+  # #include <abstractions/launchpad-integration>
   #include <abstractions/nameservice>
   #include <abstractions/private-files-strict>
   #include <abstractions/ssl_certs>
@@ -46,6 +46,7 @@
   /usr/bin/gvfs-open rmix,
   /usr/bin/pidgin r,
   /usr/bin/xdg-open rmix,
+  /usr/local/bin/tor-browser rmUx,
 
   /usr/share/gnome/applications/ r,
   /usr/share/glib-2.0/schemas/gschemas.compiled r,

