--- a/etc/apparmor.d/usr.bin.pidgin	2018-01-07 12:33:52.000000000 +0000
+++ b/etc/apparmor.d/usr.bin.pidgin	2018-01-17 12:58:45.287388991 +0000
@@ -13,7 +13,7 @@
   #include <abstractions/gnome>
   #include <abstractions/gstreamer>
   #include <abstractions/ibus>
-  #include <abstractions/launchpad-integration>
+  # #include <abstractions/launchpad-integration>
   #include <abstractions/nameservice>
   #include <abstractions/private-files-strict>
   #include <abstractions/ssl_certs>
@@ -69,6 +69,7 @@
   /usr/bin/gvfs-open rmix,
   /usr/bin/pidgin r,
   /usr/bin/xdg-open rmix,
+  /usr/local/bin/tor-browser rmUx,
 
   /etc/purple/prefs.xml r,
 
