--- a/etc/apparmor.d/usr.bin.thunderbird	2020-06-12 13:56:44.453139641 +0200
+++ b/etc/apparmor.d/usr.bin.thunderbird	2020-06-12 14:01:43.694759478 +0200
@@ -15,7 +15,6 @@
   # TODO: finetune this for required accesses
   #include <abstractions/dbus>
   #include <abstractions/dbus-accessibility>
-  #include <abstractions/dbus-session>
   #include <abstractions/dconf>
   #include <abstractions/gnome>
   #include <abstractions/ibus>
@@ -24,7 +23,6 @@
   #include <abstractions/p11-kit>
   #include <abstractions/private-files>
   #include <abstractions/ssl_certs>
-  #include <abstractions/ubuntu-browsers>
   #include <abstractions/ubuntu-browsers.d/java>
   #include <abstractions/ubuntu-helpers>
 
@@ -45,32 +43,21 @@
 
   # Allow opening attachments
   # TODO: create and use abstractions for opening various file formats
-  /{usr/local/,usr/,}bin/* Cx -> sanitized_helper,
+  /{usr/,}bin/{[^gp],g[^p],gp[^g],p[^s]}* Cx -> sanitized_helper,
+  /usr/local/bin/{[^t],t[^o],to[^r],tor[^-],tor-[^b],tor-b[^r],tor-br[^o],tor-bro[^w],tor-brow[^s],tor-brows[^e],tor-browse[^r]}* Cx -> sanitized_helper,
+  /usr/local/bin/tor-browser Uxmr,
   /usr/lib/libreoffice/program/soffice Cxr -> sanitized_helper,
 
   # Allow opening links
-  # GDesktopAppInfo in GLib 2.64.x uses a very small shell script
-  # to launch .desktop files, instead of gio-launch-desktop
-  /{usr/,}bin/{dash,bash} ixr,
   # With older GLib we might still be on the fallback code path
   # (remove this after Debian 11 and Ubuntu 20.04)
   /usr/lib/@{multiarch}/glib-[0-9]*/gio-launch-desktop ix,
 
-  # For Xubuntu to launch the browser
-  /usr/bin/exo-open ixr,
-  /usr/lib/@{multiarch}/xfce4/exo-[1-9]/exo-helper-[1-9] ixr,
-  /etc/xdg/xdg-xubuntu/xfce4/helpers.rc r,
-  /etc/xdg/xfce4/helpers.rc r,
-  owner @{HOME}/.config/xfce4/helpers.rc r,
-
   # for crash reports?
   ptrace (read,trace) peer=@{profile_name},
 
   /usr/lib/thunderbird/thunderbird{,-bin} ixr,
 
-  # Pulseaudio
-  /usr/bin/pulseaudio Pixr,
-
   owner @{HOME}/.{cache,config}/dconf/user rw,
   owner @{HOME}/.cache/thumbnails/** r,
   owner /run/user/[0-9]*/dconf/user rw,
@@ -146,6 +133,10 @@
   deny /boot/vmlinuz* r,
   deny /var/cache/fontconfig/ w,
 
+  # needed for GNOME Shell's on-screen keyboard
+  # Tails-specific: needed only because we disable the dbus-session abstraction
+  /etc/machine-id r,
+
   # noisy file dialog:
   #
   # TODO: remove these rules when file dialogs becomes "trusted helpers" that can
@@ -270,7 +261,6 @@
   /etc/lsb-release r,
   /etc/ssl/openssl.cnf r,
   /usr/lib/thunderbird/crashreporter ix,
-  /usr/bin/expr ix,
   /sys/devices/system/cpu/ r,
   /sys/devices/system/cpu/** r,
 
@@ -430,4 +420,3 @@
   # Site-specific additions and overrides. See local/README for details.
   #include <local/usr.bin.thunderbird>
 }
-
