diff --git a/profiles/apparmor.d/abstractions/python b/profiles/apparmor.d/abstractions/python
index 2ec2a4b7..f99ae64d 100644
--- a/etc/apparmor.d/abstractions/python
+++ b/etc/apparmor.d/abstractions/python
@@ -28,7 +28,7 @@
   /{var,usr}/lib/{pyshared,pycentral,python-support}/**  r,
   /usr/lib/{pyshared,pycentral,python-support}/**.so     mr,
   /var/lib/{pyshared,pycentral,python-support}/**.pyc    mr,
-  /usr/lib/python3/dist-packages/**.so                   mr,
+  /usr/{,local/}lib/python3/dist-packages/**.so          mr,
 
   # wx paths
   /usr/lib/wx/python/*.pth r,
-- 
2.18.0

