--- a/etc/apparmor.d/abstractions/user-tmp	2012-07-17 17:30:16.000000000 +0000
+++ b/etc/apparmor.d/abstractions/user-tmp	2014-09-17 05:39:57.871556000 +0000
@@ -14,7 +14,7 @@
   owner @{HOME}/tmp/    rw,
 
   # global tmp directories
-  owner /var/tmp/**     rwkl,
-  /var/tmp/             rw,
-  owner /tmp/**         rwkl,
-  /tmp/                 rw,
+  owner /{,lib/live/mount/overlay/}var/tmp/**     rwkl,
+  /{,lib/live/mount/overlay/}var/tmp/             rw,
+  owner /{,lib/live/mount/overlay/}tmp/**         rwkl,
+  /{,lib/live/mount/overlay/}tmp/                 rw,

