--- a/etc/apparmor.d/tunables/home	2012-07-17 17:30:16.000000000 +0000
+++ b/etc/apparmor.d/tunables/home	2014-09-17 05:23:26.383556000 +0000
@@ -18,7 +18,7 @@
 # @{HOMEDIRS} is a space-separated list of where user home directories
 # are stored, for programs that must enumerate all home directories on a
 # system.
-@{HOMEDIRS}=/home/
+@{HOMEDIRS}=/home/ /lib/live/mount/overlay/home/
 
 # Also, include files in tunables/home.d for site-specific adjustments to
 # @{HOMEDIRS}.

