diff -Naur chroot.orig/usr/share/xul-ext/monkeysphere/install.rdf chroot/usr/share/xul-ext/monkeysphere/install.rdf
--- chroot.orig/usr/share/xul-ext/monkeysphere/install.rdf	2011-11-29 23:48:51.984537512 +0100
+++ chroot/usr/share/xul-ext/monkeysphere/install.rdf	2011-11-29 23:49:11.792752047 +0100
@@ -39,7 +39,7 @@
       <Description>
 	<em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
 	<em:minVersion>3.5</em:minVersion>
-	<em:maxVersion>4.0.*</em:maxVersion>
+	<em:maxVersion>9.0.*</em:maxVersion>
       </Description>
     </em:targetApplication>
 
