diff -Naur chroot.orig/etc/iceweasel/profile/localstore.rdf chroot/etc/iceweasel/profile/localstore.rdf
--- chroot.orig/etc/iceweasel/profile/localstore.rdf	2014-01-17 20:10:27.078045792 +0100
+++ chroot/etc/iceweasel/profile/localstore.rdf	2014-01-17 20:13:45.012356238 +0100
@@ -6,4 +6,16 @@
 <RDF:RDF
      xmlns:NC="http://home.netscape.com/NC-rdf#"
      xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul">
+    <NC:persist RDF:resource="chrome://browser/content/browser.xul#addon-bar"/>
+    <NC:persist RDF:resource="chrome://browser/content/browser.xul#nav-bar"/>
+    <NC:persist RDF:resource="chrome://browser/content/browser.xul#navigator-toolbox"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul#addon-bar"
+                   collapsed="true" />
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul#navigator-toolbox"
+                   abp-iconposition="visible,addon-bar,addonbar-closebutton"
+                   iconsize="small" />
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul#nav-bar"
+                   iconsize="small" />
 </RDF:RDF>
