diff -Naur orig/etc/dhcp/dhclient.conf new/etc/dhcp/dhclient.conf
--- orig/etc/dhcp/dhclient.conf	2014-07-31 22:31:11.363605131 +0200
+++ new/etc/dhcp/dhclient.conf	2014-07-31 22:31:43.535349519 +0200
@@ -12,7 +12,8 @@
 
 option rfc3442-classless-static-routes code 121 = array of unsigned integer 8;
 
-send host-name = gethostname();
+#send host-name = gethostname();
+supersede host-name "amnesia";
 request subnet-mask, broadcast-address, time-offset, routers,
 	domain-name, domain-name-servers, domain-search, host-name,
 	dhcp6.name-servers, dhcp6.domain-search, dhcp6.fqdn, dhcp6.sntp-servers,

