--- chroot.orig/usr/share/xul-ext/foxyproxy-standard/chrome/content/overlay.js	2012-08-19 14:03:51.573785849 +0000
+++ chroot/usr/share/xul-ext/foxyproxy-standard/chrome/content/overlay.js	2012-08-19 14:06:19.219919480 +0000
@@ -197,6 +197,10 @@
     var contDoc = window.content.document;
     if (contDoc.documentURI.indexOf("about:neterror?e=proxyConnectFailure") ===
       0) {
+      // This function adds a suggestion to use FoxyProxy's premium proxies,
+      // which is terrible advice for Tails users. Hence we disable it.
+      return;
+
       // As we do not have a separate listener for each tab we check manually
       // whether the list element got already injected. If so we do not need to
       // add another one.
