--- chroot.orig/etc/init.d/pulseaudio	2013-04-24 19:46:38.000000000 +0000
+++ chroot/etc/init.d/pulseaudio	2014-10-12 19:24:46.242884000 +0000
@@ -5,8 +5,8 @@
 # Required-Stop:     $remote_fs $syslog
 # Should-Start:      udev network-manager
 # Should-Stop:       udev network-manager
-# Default-Start:     2 3 4 5
-# Default-Stop:      0 1 6
+# Default-Start:     
+# Default-Stop:      
 # Short-Description: Start the PulseAudio sound server
 # Description:       System mode startup script for
 #                    the PulseAudio sound server.
