--- chroot.orig/etc/init.d/plymouth	2012-06-30 10:31:21.000000000 +0000
+++ chroot/etc/init.d/plymouth	2013-01-10 10:45:41.234167902 +0000
@@ -5,9 +5,9 @@
 # Required-Start:	udev $remote_fs $all
 # Required-Stop:	$remote_fs
 # Should-Start:		$x-display-manager
-# Should-Stop:		$x-display-manager
+# Should-Stop:		
 # Default-Start:	2 3 4 5
-# Default-Stop:		0 6
+# Default-Stop:		
 # Short-Description:	Stop plymouth during boot and start it on shutdown
 ### END INIT INFO
 
