--- chroot.orig/usr/share/perl5/Tails/Persistence/Configuration/Button.pm     2014-10-03 08:33:03.131999033 +0000
+++ chroot/usr/share/perl5/Tails/Persistence/Configuration/Button.pm  2014-10-03 08:36:31.507998939 +0000
@@ -80,6 +80,7 @@
     $theme->append_search_path('/usr/share/pixmaps/cryptui/48x48');
     $theme->append_search_path('/usr/share/pixmaps/seahorse/48x48');
     $theme->append_search_path('/usr/share/icons/gnome-colors-common/32x32/apps/');
+    $theme->append_search_path('/usr/share/app-install/icons/');
 
     return $theme;
  }
--- chroot.orig/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm	2014-10-03 08:33:03.131999033 +0000
+++ chroot/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm	2014-10-03 08:36:31.507998939 +0000
@@ -135,6 +135,16 @@
             icon_name   => 'printer',
         },
         {
+            name        => $self->encoding->decode(gettext(q{Bitcoin client})),
+            description => $self->encoding->decode(gettext(
+                q{Electrum's bitcoin wallet and configuration}
+            )),
+            destination => '/home/amnesia/.electrum',
+            options     => [ 'source=electrum' ],
+            enabled     => 0,
+            icon_name   => 'electrum',
+        },
+        {
             name        => $self->encoding->decode(gettext(q{APT Packages})),
             description => $self->encoding->decode(gettext(
                 q{Packages downloaded by APT}
