diff --git a/usr/share/perl5/Tails/IUK/Frontend.pm b/usr/share/perl5/Tails/IUK/Frontend.pm
index 2f31944..e1302c2 100644
--- a/usr/share/perl5/Tails/IUK/Frontend.pm
+++ b/usr/share/perl5/Tails/IUK/Frontend.pm
@@ -56,7 +56,9 @@ class_type 'Tails::RunningSystem';
 =cut
 
 has "override_$_" => lazy_build ro Str
-    for (qw{baseurl build_target dev_dir os_release_file proc_dir run_dir},
+    for (qw{baseurl build_target dev_dir},
+         qw{os_release_file initial_install_os_release_file},
+         qw{proc_dir run_dir},
          qw{trusted_gnupg_homedir});
 
 has batch => lazy_build ro Bool;
@@ -70,7 +72,7 @@ has 'running_system' =>
     lazy_build ro 'Tails::RunningSystem',
     handles => [
         qw{upgrade_description_file_url upgrade_description_sig_url},
-        qw{product_name product_version build_target channel}
+        qw{product_name initial_install_version build_target channel}
     ];
 
 has 'free_space' =>
@@ -91,7 +93,8 @@ method _build_batch { 0; }
 
 method _build_running_system {
     my @args;
-    for (qw{baseurl build_target dev_dir liveos_mountpoint os_release_file},
+    for (qw{baseurl build_target dev_dir liveos_mountpoint},
+         qw{os_release_file initial_install_os_release_file},
          qw{proc_dir run_dir}) {
         my $attribute = "override_$_";
         my $predicate = "has_$attribute";
@@ -198,7 +201,7 @@ method checked_upgrades_file () {
 
 method get_upgrade_description () {
     my @args;
-    for (qw{baseurl build_target os_release_file}) {
+    for (qw{baseurl build_target os_release_file initial_install_os_release_file}) {
         my $attribute = "override_$_";
         my $predicate = "has_$attribute";
         if ($self->$predicate) {
diff --git a/usr/share/perl5/Tails/IUK/Install.pm b/usr/share/perl5/Tails/IUK/Install.pm
index d1f5b3f..3dbccf6 100644
--- a/usr/share/perl5/Tails/IUK/Install.pm
+++ b/usr/share/perl5/Tails/IUK/Install.pm
@@ -157,21 +157,26 @@ method space_available {
     space_available($self->liveos_mountpoint->stringify);
 }
 
+method delete_obsolete_squashfs_diffs {
+    my @keep = (
+        file($self->liveos_mountpoint->subdir('live'), 'filesystem.squashfs'),
+        file($self->liveos_mountpoint->subdir('live'), 'Tails.module'),
+        map {file($self->liveos_mountpoint, $_)} $self->all_installed_squashfs,
+    );
+
+    for my $candidate ($self->liveos_mountpoint->subdir('live')->children) {
+        next if grep { "$candidate" eq "$_" } @keep;
+        run_as_root('rm', '--force', "$candidate");
+    }
+}
+
 method upgrade_modules_file {
     my @installed_squashfs = $self->all_installed_squashfs;
 
-    my $append_str = join("\n", map { file($_)->basename } @installed_squashfs);
-    if (length($append_str)) {
+    my $new_squashfs_str = join("\n", map { file($_)->basename } @installed_squashfs);
+    if (length($new_squashfs_str)) {
         my ($temp_fh, $temp_file) = tempfile;
-        copy($self->modules_file->stringify, $temp_fh)
-            or $self->fatal(sprintf(
-                "Could not copy modules file ('%s') to temporary file ('%s')",
-                $self->modules_file, $temp_file,
-            ));
-        close $temp_fh;
-
-        $temp_fh = file($temp_file)->open('a');
-        say $temp_fh $append_str;
+        say $temp_fh 'filesystem.squashfs', "\n", $new_squashfs_str;
         close $temp_fh;
         system('sync');
 
@@ -222,6 +227,8 @@ method run {
 
     $self->upgrade_modules_file;
 
+    $self->delete_obsolete_squashfs_diffs;
+
     # upgrade syslinux' ldlinux.sys
     my $syslinux = file($self->liveos_mountpoint, qw{utils linux syslinux});
     run_as_root($syslinux, qw{-d syslinux}, $self->system_partition_file)
diff --git a/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile.pm b/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile.pm
index 0139fc1..450e206 100644
--- a/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile.pm
+++ b/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile.pm
@@ -34,7 +34,7 @@ use YAML::Any;
 =cut
 
 has "$_"    => required ro Str, predicate => "has_$_"
-    for (qw{product_name product_version build_target channel});
+    for (qw{product_name initial_install_version build_target channel});
 
 has upgrades =>
     lazy_build ro ArrayRef,
@@ -87,7 +87,7 @@ sub new_from_text {
     my $data = YAML::Any::Load($text);
 
     my %args;
-    foreach my $key (qw{product-name product-version channel build-target upgrades}) {
+    foreach my $key (qw{product-name initial-install-version channel build-target upgrades}) {
         next unless exists $data->{$key};
         my $attribute = $key; $attribute =~ s{-}{_}xmsg;
         $args{$attribute} = $data->{$key};
@@ -142,12 +142,14 @@ method full_upgrade_path () {
 method stringify () {
     my %data;
 
-    foreach my $attribute (qw{product_name product_version channel build_target upgrades}) {
+    foreach my $attribute (qw{product_name initial_install_version channel build_target upgrades}) {
         my $predicate = "has_$attribute";
         next unless $self->$predicate;
         my $key = $attribute; $key =~ s{_}{-}xmsg;
         $data{$key} = $self->$attribute;
     }
+    # for backward compatibility with older versions of the Upgrader
+    $data{'product-version'} = $data{'initial-install-version'};
 
     YAML::Any::Dump(\%data);
 }
diff --git a/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Download.pm b/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Download.pm
index 2459f6a..4d354f8 100644
--- a/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Download.pm
+++ b/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Download.pm
@@ -43,12 +43,14 @@ class_type 'Tails::RunningSystem';
 
 has "$_" => lazy_build ro Str
     for (qw{override_baseurl override_build_target override_os_release_file
+            override_initial_install_os_release_file
             trusted_gnupg_homedir});
 has 'running_system' =>
     lazy_build ro 'Tails::RunningSystem',
     handles => [
         qw{upgrade_description_file_url upgrade_description_sig_url},
-        qw{product_name product_version build_target channel}
+        qw{product_name initial_install_version build_target},
+        qw{channel}
     ];
 
 
@@ -64,7 +66,7 @@ method _build_trusted_gnupg_homedir     {
 
 method _build_running_system {
     my @args;
-    for (qw{baseurl build_target os_release_file}) {
+    for (qw{baseurl build_target os_release_file initial_install_os_release_file}) {
         my $attribute = "override_$_";
         my $predicate = "has_$attribute";
         if ($self->$predicate) {
@@ -83,12 +85,15 @@ method matches_running_system ($description_str) {
     assert(defined $description_str);
     my $description = YAML::Any::Load($description_str);
     assert_hashref($description);
-    foreach (qw{product_name product_version build_target channel}) {
+    foreach (qw{product_name initial_install_version build_target channel}) {
         my $accessor = my $field = $_;
         $field =~ s{_}{-}gxms;
         exists  $description->{$field}             or return;
         defined $description->{$field}             or return;
-        $description->{$field} eq $self->$accessor or return;
+        if ($description->{$field} ne $self->$accessor) {
+            warn "Expected for fied $field: ", $self->$accessor, ", got: ", $description->{$field};
+            return;
+        }
     }
     return 1;
 }
diff --git a/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Generate.pm b/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Generate.pm
index d4be264..fb904cc 100644
--- a/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Generate.pm
+++ b/usr/share/perl5/Tails/IUK/UpgradeDescriptionFile/Generate.pm
@@ -95,10 +95,10 @@ method run {
 
 method create_udf_for_new_release ($channel) {
     my $description = Tails::IUK::UpgradeDescriptionFile->new(
-        product_name    => $self->product_name,
-        product_version => $self->version,
-        build_target    => $self->build_target,
-        channel         => $channel,
+        product_name            => $self->product_name,
+        initial_install_version => $self->version,
+        build_target            => $self->build_target,
+        channel                 => $channel,
     );
     $self->udf_for_new_release($channel)->parent->mkpath;
     $self->udf_for_new_release($channel)->spew(
@@ -110,10 +110,10 @@ method create_udf_for_next_release ($version, $channel) {
     my $udf = $self->udf_for($version, channel => $channel);
 
     my $description = Tails::IUK::UpgradeDescriptionFile->new(
-        product_name    => $self->product_name,
-        product_version => $version,
-        build_target    => $self->build_target,
-        channel         => $channel,
+        product_name            => $self->product_name,
+        initial_install_version => $version,
+        build_target            => $self->build_target,
+        channel                 => $channel,
     );
     $udf->parent->mkpath;
     $udf->spew($description->stringify);
@@ -128,10 +128,10 @@ method update_udf_for_previous_release ($previous_version) {
     }
     else {
         $description = Tails::IUK::UpgradeDescriptionFile->new(
-            product_name    => $self->product_name,
-            product_version => $previous_version,
-            build_target    => $self->build_target,
-            channel         => $self->channel,
+            product_name            => $self->product_name,
+            initial_install_version => $previous_version,
+            build_target            => $self->build_target,
+            channel                 => $self->channel,
         );
         $udf->parent->mkpath;
     }
diff --git a/usr/share/perl5/Tails/RunningSystem.pm b/usr/share/perl5/Tails/RunningSystem.pm
index 5a6ca52..1587758 100644
--- a/usr/share/perl5/Tails/RunningSystem.pm
+++ b/usr/share/perl5/Tails/RunningSystem.pm
@@ -43,9 +43,11 @@ has 'upgrade_description_url_schema_version' => lazy_build ro Int;
 
 has "$_" => lazy_build ro Str
     for (
-        qw{baseurl product_name product_version build_target channel},
+        qw{baseurl product_name initial_install_version product_version},
+        qw{build_target channel},
         qw{upgrade_description_file_url upgrade_description_sig_url},
-        qw{dev_dir os_release_file proc_dir run_dir},
+        qw{dev_dir os_release_file initial_install_os_release_file},
+        qw{proc_dir run_dir},
     );
 
 has 'udisks' =>
@@ -105,10 +107,19 @@ has 'override_started_from_device_installed_with_tails_installer' =>
 method _build_upgrade_description_url_schema_version { 1 }
 method _build_dev_dir         { '/dev' }
 method _build_os_release_file { '/etc/os-release' }
+method _build_initial_install_os_release_file {
+    '/lib/live/mount/rootfs/filesystem.squashfs/etc/os-release'
+}
 method _build_proc_dir        { '/proc' }
 method _build_run_dir         { '/var/run' }
 method _build_product_name    { $self->os_release_get('TAILS_PRODUCT_NAME') }
 method _build_product_version { $self->os_release_get('TAILS_VERSION_ID')   }
+method _build_initial_install_version {
+    $self->os_release_get(
+        'TAILS_VERSION_ID',
+        'os_release_file' => $self->initial_install_os_release_file,
+    )
+}
 method _build_baseurl         { 'https://tails.boum.org' }
 method _build_udisks          { Tails::UDisks->new(); }
 
@@ -128,7 +139,7 @@ method _build_upgrade_description_file_url {
         $self->baseurl,
         $self->upgrade_description_url_schema_version,
         $self->product_name,
-        $self->product_version,
+        $self->initial_install_version,
         $self->build_target,
         $self->channel,
     );
@@ -251,11 +262,14 @@ http://www.freedesktop.org/software/systemd/man/os-release.html
 Throws an exception if not found.
 
 =cut
-method os_release_get ($key) {
-    assert(-e $self->os_release_file);
+method os_release_get ($key, :$os_release_file?) {
+    my $file = defined $os_release_file
+        ? $os_release_file
+        : $self->os_release_file;
+    assert(-e $file);
     assert_like($key, qr{[_A-Z]+});
 
-    my $fh = file($self->os_release_file)->openr;
+    my $fh = file($file)->openr;
 
     while (<$fh>) {
         chomp;
@@ -266,7 +280,7 @@ method os_release_get ($key) {
 
     croak(sprintf(
         "Could not retrieve value of '%s' in '%s'",
-        $key, $self->os_release_file,
+        $key, $file,
     ));
 }
 
