--- a/etc/gdm3/greeter.dconf-defaults	2014-11-17 11:16:23.916000000 +0000
+++ b/etc/gdm3/greeter.dconf-defaults	2014-11-17 11:16:51.728000000 +0000
@@ -33,3 +33,10 @@
 # - Show a login welcome message
 # banner-message-enable=true
 # banner-message-text='Welcome'
+
+[org/gnome/desktop/session]
+session-name='gdm-tails'
+
+[org/gnome/desktop/background]
+picture-uri='file:///usr/share/tails/desktop_wallpaper.png'
+

