--- chroot.orig/etc/gdm3/greeter.gsettings	2014-05-11 21:58:13.547765234 +0200
+++ chroot/etc/gdm3/greeter.gsettings	2014-05-11 22:00:46.493721796 +0200
@@ -17,6 +17,10 @@
 # picture-options='none'
 # primary-color='#000000'
 
+[org.gnome.desktop.background]
+picture-options='none'
+primary-color='#204A87'
+
 # Greeter session choice
 # ======================
 # Use 'gdm-shell' for the GNOME Shell version.
