--- a/usr/share/pyshared/tailsgreeter/persistence.py	2012-12-31 18:21:00.324004572 +0000
+++ b/usr/share/pyshared/tailsgreeter/persistence.py	2012-12-31 18:21:17.308004822 +0000
@@ -45,7 +45,7 @@
             [
                 "/usr/bin/sudo", "-n", "/usr/local/sbin/live-persist",
                 "--log-file=/var/log/live-persist",
-                "--encryption=luks", "--media=removable",
+                "--encryption=luks",
                 "list", "TailsData"
             ],
             stdout=subprocess.PIPE,
