diff --git a/gdm/PostLogin.default b/gdm/PostLogin.default
index 0d35c21..ada42a8 100755
--- a/etc/gdm3/PostLogin/Default
+++ b/etc/gdm3/PostLogin/Default
@@ -71,7 +71,8 @@ fi
 
 ### Physical security
 log "Running /usr/local/lib/tails-unblock-network..."
-/usr/local/lib/tails-unblock-network &
+/usr/local/lib/tails-unblock-network
+log "tails-unblock-network has exited (status=$?)."
 
 ### Localization
 
