diff --git a/config/gdm-tails.json b/config/gdm-tails.json
index 801617c..dcaac5e 100644
--- a/usr/share/gnome-shell/modes/gdm-tails.json
+++ b/usr/share/gnome-shell/modes/gdm-tails.json
@@ -9,7 +9,7 @@
     "panel": {
         "left": [],
         "center": ["dateMenu"],
-        "right": ["a11yGreeter", "keyboard", "aggregateMenu"]
+        "right": ["a11y", "keyboard", "aggregateMenu"]
     },
     "panelStyle": "loginScreen",
     "stylesheetName": "gnome-classic.css"
diff --git a/config/gdm-tails.session b/config/gdm-tails.session
index 11ad3ee..f5f4f7e 100644
--- a/usr/share/gnome-session/sessions/gdm-tails.session
+++ b/usr/share/gnome-session/sessions/gdm-tails.session
@@ -1,3 +1,3 @@
 [GNOME Session]
 Name=Display Manager
-RequiredComponents=gdm-shell-tails;tails-greeter;gnome-settings-daemon;
+RequiredComponents=gdm-shell-tails;tails-greeter;org.gnome.SettingsDaemon.A11ySettings;org.gnome.SettingsDaemon.Clipboard;org.gnome.SettingsDaemon.Color;org.gnome.SettingsDaemon.Datetime;org.gnome.SettingsDaemon.Housekeeping;org.gnome.SettingsDaemon.Keyboard;org.gnome.SettingsDaemon.MediaKeys;org.gnome.SettingsDaemon.Mouse;org.gnome.SettingsDaemon.Power;org.gnome.SettingsDaemon.PrintNotifications;org.gnome.SettingsDaemon.Rfkill;org.gnome.SettingsDaemon.ScreensaverProxy;org.gnome.SettingsDaemon.Sharing;org.gnome.SettingsDaemon.Smartcard;org.gnome.SettingsDaemon.Sound;org.gnome.SettingsDaemon.Wacom;org.gnome.SettingsDaemon.XSettings;
diff --git a/tailsgreeter/gui.py b/tailsgreeter/gui.py
index 745429a..2cacc33 100644
--- a/usr/lib/python3/dist-packages/tailsgreeter/gui.py
+++ b/usr/lib/python3/dist-packages/tailsgreeter/gui.py
@@ -883,9 +883,6 @@ class GreeterMainWindow(Gtk.Window, TranslatableWindow):
 
     def finish_login(self):
         logging.info("Starting the session")
-        # /usr/lib/systemd/user/tails-greeter-session-helper.service will
-        # restore the cursor to the left pointer state.
-        self.get_root_window().set_cursor(Gdk.Cursor.new(Gdk.CursorType.WATCH))
         self.greeter.login()
         return False
