--- chroot.orig/etc/iceweasel/profile/localstore.rdf	2012-11-23 17:23:33.000000000 +0000
+++ chroot/etc/iceweasel/profile/localstore.rdf	2012-12-12 15:27:38.014642266 +0000
@@ -2,4 +2,12 @@
 <RDF:RDF
      xmlns:NC="http://home.netscape.com/NC-rdf#"
      xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul">
+    <NC:persist RDF:resource="chrome://browser/content/browser.xul#addon-bar"/>
+    <NC:persist RDF:resource="chrome://browser/content/browser.xul#navigator-toolbox"/>
+  </RDF:Description>
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul#addon-bar"
+                   collapsed="true" />
+  <RDF:Description RDF:about="chrome://browser/content/browser.xul#navigator-toolbox"
+                   abp-iconposition="visible,addon-bar,addonbar-closebutton" />
 </RDF:RDF>
