--- /usr/share/xul-ext/torbirdy/chrome/content/emailwizard.xul.orig	2015-09-28 00:28:45.164177872 +0000
+++ /usr/share/xul-ext/torbirdy/chrome/content/emailwizard.xul	2015-09-28 00:28:55.400376965 +0000
@@ -22,10 +22,10 @@
 
         <menulist id="torbirdy-protocol">
           <menupopup>
-            <menuitem label="POP3"
-                      value="pop3" />
             <menuitem label="IMAP"
                       value="imap" />
+            <menuitem label="POP3"
+                      value="pop3" />
           </menupopup>
         </menulist>
 
