--- a/etc/default/intel-microcode.orig	2013-10-19 22:09:10.166251353 +0200
+++ b/etc/default/intel-microcode	2013-10-19 22:09:37.438714869 +0200
@@ -10,12 +10,12 @@
 # no:      do not install microcode to the initramfs
 # early:   use early firmware (requires kernel support)
 # yes:     use normal initramfs (too late to repair the worst problems)
-#IUCODE_TOOL_INITRAMFS=
+IUCODE_TOOL_INITRAMFS=yes
 
 # Should we use "iucode_tool --scan-system" to reduce the initramfs
 # bloat, by installing only microcode appropriate to the system
 # processors active on the system where update-initramfs is run?
-IUCODE_TOOL_SCANCPUS=yes
+IUCODE_TOOL_SCANCPUS=no
 
 # Extra options to pass to iucode_tool, useful to forbid or to
 # force the inclusion of microcode for specific processor signatures.
