diff --git a/usr/share/perl5/Tails/IUK/Frontend.pm b/usr/share/perl5/Tails/IUK/Frontend.pm
index f9366f6..6c8ad15 100644
--- a/usr/share/perl5/Tails/IUK/Frontend.pm
+++ b/usr/share/perl5/Tails/IUK/Frontend.pm
@@ -239,7 +239,7 @@ method no_incremental_explanation (Str $no_incremental_reason) {
     }
     elsif ($no_incremental_reason eq 'not-installed-with-tails-installer') {
         $explanation = gettext(
-            q{your device was not created using Tails Installer}
+            q{your device was not created using a USB image or Tails Installer}
         );
     }
     elsif ($no_incremental_reason eq 'non-writable-device') {
