diff --git a/usr/share/perl5/Tails/IUK/Frontend.pm b/usr/share/perl5/Tails/IUK/Frontend.pm
index 7d14688..62a2ac4 100644
--- a/usr/share/perl5/Tails/IUK/Frontend.pm
+++ b/usr/share/perl5/Tails/IUK/Frontend.pm
@@ -169,7 +169,7 @@ method dialog (Str $question,
     if ($type ne 'question') {
         assert_undefined($cancel_label);
     }
-    my @cmd  = ('zenity', "--$type", '--text', $question);
+    my @cmd  = ('zenity', "--$type", '--ellipsize', '--text', $question);
     my $info = $question;
     if (defined $title) {
         $info = "$title\n$info";
