[0m[01;34mTails[0m/
diff --git a/usr/share/perl5/Tails/IUK/Frontend.pm b/usr/share/perl5/Tails/IUK/Frontend.pm
index 2f31944..dbf34dd 100644
--- a/usr/share/perl5/Tails/IUK/Frontend.pm
+++ b/usr/share/perl5/Tails/IUK/Frontend.pm
@@ -13,7 +13,7 @@ use MooseX::Types::Moose qw( :all );
 use MooseX::Types::Path::Class;
 use MooseX::Has::Sugar::Saccharin;
 
-with 'Tails::IUK::Role::HasEncoding';
+with 'Tails::Role::HasEncoding';
 with 'MooseX::Getopt::Dashes';
 
 use 5.10.1;
@@ -28,7 +28,6 @@ use Data::Dumper;
 use English qw{-no_match_vars};
 use Env;
 use IPC::Run qw{run start};
-use IPC::Run::SafeHandles;
 use Number::Format qw(:subs);
 use Path::Class;
 use String::Errf qw{errf};
diff --git a/usr/share/perl5/Tails/IUK/Install.pm b/usr/share/perl5/Tails/IUK/Install.pm
index d1f5b3f..55f9b14 100644
--- a/usr/share/perl5/Tails/IUK/Install.pm
+++ b/usr/share/perl5/Tails/IUK/Install.pm
@@ -80,7 +80,7 @@ has 'running_system' =>
 method _build_tempdir {
     $self->remount_liveos_rw;
 
-    my $base_live_tmp_dir = $self->liveos_mountpoint->subdir('tmp')->stringify;
+    my $base_live_tmp_dir = $self->liveos_mountpoint->child('tmp')->stringify;
 
     # --parents is used to avoid error if existing, not to create parents.
     run_as_root('mkdir', '--parents', $base_live_tmp_dir);
diff --git a/usr/share/perl5/Tails/IUK/Role/HasCodeset.pm b/usr/share/perl5/Tails/IUK/Role/HasCodeset.pm
deleted file mode 100644
index 0f1721d..0000000
--- a/usr/share/perl5/Tails/IUK/Role/HasCodeset.pm
+++ /dev/null
@@ -1,38 +0,0 @@
-=head1 NAME
-
-Tails::IUK::Role::HasCodeset - role to get the codeset being used
-
-=head1 SYNOPSIS
-
-    See Tails::IUK::Frontend for a real-life usage example.
-
-=cut
-
-package Tails::IUK::Role::HasCodeset;
-use Moose::Role;
-
-
-
-use namespace::autoclean;
-use Try::Tiny;
-
-has 'codeset'  => (
-    isa        => 'Str',
-    is         => 'ro',
-    lazy_build => 1,
-);
-
-sub _build_codeset {
-    my $codeset;
-    try {
-        require I18N::Langinfo;
-        I18N::Langinfo->import(qw(langinfo CODESET));
-        $codeset = langinfo(CODESET());
-    } catch {
-        die "No default character code set configured.\nPlease fix your locale settings.";
-    };
-    $codeset;
-}
-
-no Moose::Role;
-1; # End of Tails::IUK::Role::HasCodeset
diff --git a/usr/share/perl5/Tails/IUK/Role/HasEncoding.pm b/usr/share/perl5/Tails/IUK/Role/HasEncoding.pm
deleted file mode 100644
index 7d2d5fc..0000000
--- a/usr/share/perl5/Tails/IUK/Role/HasEncoding.pm
+++ /dev/null
@@ -1,36 +0,0 @@
-=head1 NAME
-
-Tails::IUK::Role::HasEncoding - role to provide an Encode::Encoding objet
-
-=head1 SYNOPSIS
-
-    See Tails::IUK::Frontend for a real-life usage example.
-
-=cut
-
-package Tails::IUK::Role::HasEncoding;
-use Moose::Role;
-
-
-
-use namespace::autoclean;
-with 'Tails::IUK::Role::HasCodeset';
-use Encode qw{find_encoding};
-use Moose::Util::TypeConstraints;
-
-class_type('Encode::Encoding');
-class_type('Encode::XS');
-
-has 'encoding' => (
-    isa        => 'Encode::Encoding | Encode::XS',
-    is         => 'ro',
-    lazy_build => 1,
-);
-
-sub _build_encoding {
-    my $self = shift;
-    find_encoding($self->codeset);
-}
-
-no Moose::Role;
-1; # End of Tails::IUK::Role::HasEncoding
