--- a/usr/share/perl5/Desktop/Notify/Notification.pm
+++ b/usr/share/perl5/Desktop/Notify/Notification.pm
@@ -83,7 +83,7 @@ sub show {
                  $self->{server}->{app_icon},
                  $self->{summary},
                  $self->{body},
-                 [%{$self->{actions}}],
+                 $self->{actions},
                  $self->{hints},
                  $self->{timeout} || 0,
                 );
