[0m[01;34mTails[0m/  Tails.pm
diff --git a/usr/share/perl5/Tails/RunningSystem.pm b/usr/share/perl5/Tails/RunningSystem.pm
index 9409ad4..c4b6905 100644
--- a/usr/share/perl5/Tails/RunningSystem.pm
+++ b/usr/share/perl5/Tails/RunningSystem.pm
@@ -144,7 +144,7 @@ method _build_constants {
 
 sub _build_liveos_mountpoint {
     my $self = shift;
-    dir('/lib/live/mount/medium');
+    dir('/run/live/medium');
 }
 
 sub _build_boot_block_device {
