[0m[01;34mTails[0m/  Tails.pm
diff --git a/usr/share/perl5/Tails/RunningSystem.pm b/usr/share/perl5/Tails/RunningSystem.pm
index 9409ad4..c4b6905 100644
--- a/usr/share/perl5/Tails/RunningSystem.pm
+++ b/usr/share/perl5/Tails/RunningSystem.pm
@@ -177,7 +177,7 @@
 }

 method _build_liveos_mountpoint () {
-    path('/lib/live/mount/medium');
+    path('/run/live/medium');
 }

 method _build_boot_block_device () {
