--- chroot.orig/usr/share/polkit-1/actions/org.freedesktop.udisks.policy.bak	2013-10-02 14:49:35.476000000 +0000
+++ chroot/usr/share/polkit-1/actions/org.freedesktop.udisks.policy	2013-10-02 15:12:56.544000000 +0000
@@ -140,7 +140,7 @@
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin</allow_active>
     </defaults>
   </action>
 

