--- a/usr/share/applications/tails-persistence-delete.desktop	2014-10-15 15:26:55.000000000 +0000
+++ b/usr/share/applications/tails-persistence-delete.desktop	2014-11-26 17:13:35.000000000 +0000
@@ -7,7 +7,7 @@
 Comment[fr]=Supprimer le volume de stockage persistant et son contenu
 Comment[sv]=Ta bort den bestående lagringen och allt sparat i den
 Icon=tails-persistence-setup.png
-Exec=tails-persistence-setup --step delete
+Exec=/usr/local/bin/tails-delete-persistent-volume
 Terminal=false
 Categories=System;Tails;
 StartupNotify=false
--- a/usr/share/applications/tails-persistence-setup.desktop	2014-10-15 15:26:55.000000000 +0000
+++ b/usr/share/applications/tails-persistence-setup.desktop	2014-11-26 17:13:35.000000000 +0000
@@ -7,7 +7,7 @@
 Comment[fr]=Configurer quelles données, et la configuration de quelles applications, sont conservées d'un démarrage à l'autre
 Comment[sv]=Välj vilka filer och programinställningar som ska sparas mellan omstarter av systemet
 Icon=tails-persistence-setup.png
-Exec=tails-persistence-setup
+Exec=/usr/local/bin/tails-persistence-setup
 Terminal=false
 Categories=System;Tails;
 StartupNotify=false
