--- a/etc/init.d/kexec-load	2014-12-01 20:23:12.826065938 +0100
+++ b/etc/init.d/kexec-load	2014-12-01 20:23:31.389572352 +0100
@@ -106,19 +106,6 @@
 	exit 3
 	;;
   stop)
-	# If running systemd, we want kexec reboot only if current
-	# command is reboot. If not running systemd, check if
-	# runlevel has been changed to 6 which indicates we are
-	# rebooting
-	if [ -d /run/systemd/system ]; then
-		systemctl list-jobs systemd-kexec.service | grep -q systemd-kexec.service
-		if [ $? -ne 0 ]; then
-			exit 0
-		fi
-	else if [ -x /sbin/runlevel -a "$(runlevel | awk '{ print $2 }')" != "6" ]; then 
-			exit 0
-		fi
-	fi
 	do_stop
 	;;
   *)

