diff -Naur chroot.orig/usr/share/applications/tails-persistence-delete.desktop chroot/usr/share/applications/tails-persistence-delete.desktop
--- chroot.orig/usr/share/applications/tails-persistence-delete.desktop	2013-06-10 14:31:23.092537529 +0200
+++ chroot/usr/share/applications/tails-persistence-delete.desktop	2013-06-10 14:32:07.029151302 +0200
@@ -5,7 +5,7 @@
 Comment=Delete the persistent volume and its content
 Comment[fr]=Supprimer le volume de stockage persistant et son contenu
 Icon=tails-persistence-setup.png
-Exec=tails-persistence-setup --step delete
+Exec=/usr/local/bin/tails-delete-persistent-volume
 Terminal=false
 Categories=System;Tails;
 StartupNotify=false
diff -Naur chroot.orig/usr/share/applications/tails-persistence-setup.desktop chroot/usr/share/applications/tails-persistence-setup.desktop
--- chroot.orig/usr/share/applications/tails-persistence-setup.desktop	2013-06-10 14:30:53.880129123 +0200
+++ chroot/usr/share/applications/tails-persistence-setup.desktop	2013-06-10 14:32:00.377058414 +0200
@@ -5,7 +5,7 @@
 Comment=Configure which files and application configuration are saved between working sessions
 Comment[fr]=Configurer quelles données, et la configuration de quelles applications, sont conservées d'un démarrage à l'autre
 Icon=tails-persistence-setup.png
-Exec=tails-persistence-setup
+Exec=/usr/local/bin/tails-persistence-setup
 Terminal=false
 Categories=System;Tails;
 StartupNotify=false
