--- a/etc/init.d/kexec-load	2014-11-21 09:52:51.744000000 +0000
+++ b/etc/init.d/kexec-load	2014-11-21 09:53:08.384000000 +0000
@@ -101,15 +101,6 @@
 	exit 3
 	;;
   stop)
-	# If running systemd, we want kexec reboot only if current
-	# command is reboot
-	pidof systemd > /dev/null 2>&1
-	if [ $? -eq 0 ]; then
-		systemctl list-jobs systemd-reboot.service | grep -q systemd-reboot.service
-		if [ $? -ne 0 ]; then
-			exit 0
-		fi
-	fi
 	do_stop
 	;;
   *)

