--- a/etc/init.d/apparmor	2014-11-26 13:28:03.932000000 +0000
+++ b/etc/init.d/apparmor	2014-11-26 13:28:41.776000000 +0000
@@ -24,9 +24,9 @@
 #
 ### BEGIN INIT INFO
 # Provides: apparmor
-# Required-Start: $remote_fs
+# Required-Start:
 # Required-Stop: umountfs
-# Default-Start: S
+# Default-Start: 2 3 4 5
 # Default-Stop:
 # Short-Description: AppArmor initialization
 # Description: AppArmor init script. This script loads all AppArmor profiles.
