diff --git a/lib/Tails/IUK/Install.pm b/lib/Tails/IUK/Install.pm
index 1ab0612..20b5b63 100644
--- a/usr/share/perl5/Tails/IUK/Install.pm
+++ b/usr/share/perl5/Tails/IUK/Install.pm
@@ -173,8 +173,9 @@ method upgrade_modules_file {
         $temp_fh = file($temp_file)->open('a');
         say $temp_fh $append_str;
         close $temp_fh;
+        system('sync');
 
-        run_as_root('cp', '--force', $temp_file, $self->modules_file);
+        run_as_root('nocache', '/bin/cp', '--force', $temp_file, $self->modules_file);
     }
 }
 
