--- chroot.orig/etc/gdm3/PostLogin/Default	2014-10-09 23:57:25.319996058 +0000
+++ chroot/etc/gdm3/PostLogin/Default	2014-10-09 23:58:19.147993943 +0000
@@ -133,7 +133,7 @@
 # Choose which browser language pack to use
 LONG_LOCALE="$(echo ${TAILS_LOCALE_NAME} | sed -e 's/\..*$//' -e 's/_/-/')"
 SHORT_LOCALE="$(echo ${LONG_LOCALE} | cut -d"-" -f1)"
-LANGPACK_DIR="/usr/local/lib/tor-browser/extensions"
+LANGPACK_DIR="/usr/local/share/tor-browser-extensions"
 if [ -e "${LANGPACK_DIR}/langpack-${LONG_LOCALE}@firefox.mozilla.org.xpi" ]; then
     BROWSER_LOCALE="${LONG_LOCALE}"
 elif [ -e "${LANGPACK_DIR}/langpack-${SHORT_LOCALE}@firefox.mozilla.org.xpi" ]; then
