--- chroot.orig/usr/share/perl5/Tails/IUK/TargetFile/Download.pm	2012-09-24 15:41:46.907999736 +0000
+++ chroot/usr/share/perl5/Tails/IUK/TargetFile/Download.pm	2012-09-24 15:42:09.959998817 +0000
@@ -72,7 +72,7 @@
         SSL_verify_mode => 0,
     });
     unless ($ENV{HARNESS_ACTIVE} or $ENV{DISABLE_PROXY}) {
-        $ua->proxy([qw(http https)] => 'socks://127.0.0.1:9050');
+        $ua->proxy([qw(http https)] => 'socks://127.0.0.1:9062');
     }
     $ua->protocols_allowed([qw(http https)]);
     my $req = HTTP::Request->new('GET', $self->uri);


--- chroot.orig/usr/share/perl5/Tails/IUK/UpdateDescriptionFile/Download.pm	2012-09-24 15:41:33.472000071 +0000
+++ chroot/usr/share/perl5/Tails/IUK/UpdateDescriptionFile/Download.pm	2012-09-24 15:41:57.020000109 +0000
@@ -119,7 +119,7 @@
 method get_url ($url) {
     my $ua  = LWP::UserAgent->new();
     unless ($ENV{HARNESS_ACTIVE} or $ENV{DISABLE_PROXY}) {
-        $ua->proxy([qw(http https)] => 'socks://127.0.0.1:9050');
+        $ua->proxy([qw(http https)] => 'socks://127.0.0.1:9062');
     }
     $ua->protocols_allowed([qw{https}]);
     $ua->max_size($self->max_download_size);

