--- a/usr/share/xul-ext/torbirdy/install.rdf	2018-08-07 10:48:50.000000000 +0000
+++ b/usr/share/xul-ext/torbirdy/install.rdf	2018-10-22 15:27:09.132000000 +0000
@@ -14,7 +14,7 @@
     <Description>
       <em:id>{3550f703-e582-4d05-9a08-453d09bdfdc6}</em:id>
       <em:minVersion>52.0</em:minVersion>
-      <em:maxVersion>60.0</em:maxVersion>
+      <em:maxVersion>60.*</em:maxVersion>
     </Description>
     </em:targetApplication>
 

