--- chroot.orig/etc/wgetrc	2010-09-05 13:43:50.000000000 +0000
+++ chroot/etc/wgetrc	2011-12-02 12:31:23.566907236 +0000
@@ -76,8 +76,8 @@
 
 # You can set the default proxies for Wget to use for http, https, and ftp.
 # They will override the value in the environment.
-#https_proxy = http://proxy.yoyodyne.com:18023/
-#http_proxy = http://proxy.yoyodyne.com:18023/
+https_proxy = http://127.0.0.1:8118/
+http_proxy = http://127.0.0.1:8118/
 #ftp_proxy = http://proxy.yoyodyne.com:18023/
 
 # If you do not want to use proxy at all, set this to off.
