diff -Naur usr.orig/share/applications/claws-mail.desktop usr/share/applications/claws-mail.desktop
--- chroot.orig/usr/share/applications/claws-mail.desktop	2011-10-17 14:46:00.278241354 +0200
+++ chroot/usr/share/applications/claws-mail.desktop	2011-10-17 14:47:01.782963288 +0200
@@ -5,7 +5,7 @@
 Comment=E-Mail client
 Comment[ja]=電子メールクライアント
 Comment[es]=Cliente de correo
-Exec=claws-mail
+Exec=/usr/local/bin/torified-claws-mail
 Icon=claws-mail
 Terminal=false
 Type=Application
