--- a/usr/share/applications/liferea.desktop.orig	2015-02-26 10:14:36.644624000 +0000
+++ b/usr/share/applications/liferea.desktop	2015-02-26 10:16:50.216626591 +0000
@@ -118,7 +118,7 @@
 Keywords=news;feed;aggregator;blog;podcast;
 Keywords[ar]=أخبار;تلقيمات;مدونة;تدوين;
 Keywords[he]=חדשות;ערוץ;הזנה;פיד;מאגד;בלוג;פודקסט;
-Exec=liferea
+Exec=torsocks liferea
 Icon=liferea
 StartupNotify=true
 Terminal=false

