diff -Naur chroot.orig/usr/share/applications/seahorse.desktop chroot/usr/share/applications/seahorse.desktop
--- chroot.orig/usr/share/applications/seahorse.desktop	2013-05-15 16:46:58.172950131 +0200
+++ chroot/usr/share/applications/seahorse.desktop	2013-05-15 16:49:29.734600598 +0200
@@ -123,7 +123,7 @@
 Comment[zh_CN]=管理您的密码和加密密钥
 Comment[zh_HK]=管理密碼及密碼匙
 Comment[zh_TW]=管理密碼及金鑰
-Exec=seahorse
+Exec=torsocks seahorse
 Terminal=false
 Type=Application
 Icon=seahorse
