diff -Naur chroot.orig/usr/share/applications/gobby.desktop chroot/usr/share/applications/gobby.desktop
--- chroot.orig/usr/share/applications/gobby.desktop	2012-04-29 17:46:22.010445613 +0200
+++ chroot/usr/share/applications/gobby.desktop	2012-04-29 17:46:35.930680137 +0200
@@ -21,7 +21,7 @@
 Comment[th]=แก้ไขไฟล์ข้อความร่วมกัน
 Comment[uk]=Редагуйте текстові файли разом з іншими користувачами
 Comment[zh_TW]=協作編輯文字檔案
-Exec=gobby-0.4
+Exec=torify gobby-0.4
 Terminal=false
 Type=Application
 Icon=gobby.png
