diff -Naur chroot.orig/usr/share/applications/gobby-0.5.desktop chroot/usr/share/applications/gobby-0.5.desktop
--- chroot.orig/usr/share/applications/gobby-0.5.desktop	2012-04-29 17:28:05.351879889 +0200
+++ chroot/usr/share/applications/gobby-0.5.desktop	2012-04-29 17:28:19.656123299 +0200
@@ -23,7 +23,7 @@
 Comment[en_GB]=Edit text files collaboratively
 Comment[fr]=Éditer des fichiers texte de manière collaborative
 Comment[ja]=テキストを共同作業で編集する
-Exec=gobby-0.5
+Exec=torsocks gobby-0.5
 Terminal=false
 Type=Application
 Icon=gobby-0.5
