diff -Naur chroot.orig/usr/share/applications/gobby-0.5.desktop chroot/usr/share/applications/gobby-0.5.desktop
--- chroot.orig/usr/share/applications/gobby-0.5.desktop	2014-09-08 13:21:27.923930066 +0000
+++ chroot/usr/share/applications/gobby-0.5.desktop	2014-09-08 13:21:41.183929665 +0000
@@ -30,7 +30,7 @@
 Comment[ja]=テキストを共同作業で編集する
 Comment[pt_BR]=Editar arquivos de texto de forma colaborativa
 Comment[zh_TW]=共同編輯文字檔
-Exec=gobby-0.5
+Exec=torsocks gobby-0.5
 Terminal=false
 Type=Application
 Icon=gobby-0.5
