--- chroot.orig/usr/share/applications/gobby-0.5.desktop	2011-01-13 23:45:16.000000000 +0000
+++ chroot/usr/share/applications/gobby-0.5.desktop	2011-12-02 12:32:57.681940285 +0000
@@ -3,7 +3,7 @@
 Name[en_GB]=Gobby Collaborative Editor (0.5)
 Comment=Edit text files collaboratively
 Comment[en_GB]=Edit text files collaboratively
-Exec=gobby-0.5
+Exec=torify gobby-0.5
 Terminal=false
 Type=Application
 Icon=gobby-0.5
