diff -Naur chroot.orig/usr/share/applications/claws-mail.desktop chroot/usr/share/applications/claws-mail.desktop
--- chroot.orig/usr/share/applications/claws-mail.desktop	2012-04-29 17:10:07.545617045 +0200
+++ chroot/usr/share/applications/claws-mail.desktop	2012-04-29 17:10:25.613908607 +0200
@@ -16,7 +16,7 @@
 GenericName[pt-br]=Cliente de e-mail
 GenericName[sk]=Poštový klient
 GenericName[sv]=E-postklient
-Exec=claws-mail %u
+Exec=/usr/local/bin/torified-claws-mail
 Icon=claws-mail
 Categories=Network;Email;
 Keywords=lightweight fast x-window extensible plugin pop3 imap nntp
