--- a/usr/share/applications/seahorse.desktop.orig	2013-06-29 22:38:31.433341106 +0000
+++ b/usr/share/applications/seahorse.desktop	2013-06-29 22:38:45.473340324 +0000
@@ -164,7 +164,7 @@
 Keywords[zh_CN]=密钥环;加密;安全;签名;密码;keyring;encryption;security;sign;ssh;
 Keywords[zh_HK]=keyring;encryption;security;sign;ssh;密碼匙;加密;安全;簽署;
 Keywords[zh_TW]=keyring;encryption;security;sign;ssh;金鑰;加密;安全;簽署;
-Exec=/usr/bin/seahorse
+Exec=torsocks /usr/bin/seahorse
 Terminal=false
 Type=Application
 Icon=seahorse
