diff -Naur chroot.orig/usr/share/applications/claws-mail.desktop chroot/usr/share/applications/claws-mail.desktop
--- chroot.orig/usr/share/applications/claws-mail.desktop	2012-04-29 17:10:07.545617045 +0200
+++ chroot/usr/share/applications/claws-mail.desktop	2012-04-29 17:10:25.613908607 +0200
@@ -15,7 +15,7 @@
 GenericName[pl]=Program poczty elektronicznej
 GenericName[pt-br]=Cliente de e-mail
 GenericName[sk]=Poštový klient
-Exec=claws-mail %u
+Exec=/usr/local/bin/torified-claws-mail
 Icon=claws-mail
 Categories=GTK;Network;Email;
 Comment=Lightweight and Fast GTK+ based Mail Client
