--- a/usr/share/applications/claws-mail.desktop	2014-10-28 00:50:16.000000000 +0000
+++ b/usr/share/applications/claws-mail.desktop	2014-11-26 17:13:35.000000000 +0000
@@ -16,7 +16,7 @@
 GenericName[pt-br]=Cliente de e-mail
 GenericName[sk]=Poštový klient
 GenericName[sv]=E-postklient
-Exec=claws-mail %u
+Exec=/usr/local/bin/torified-claws-mail
 Icon=claws-mail
 Categories=Network;Email;
 Keywords=lightweight;fast;gui;extensible;plugin;pop;pop3;imap;imap4;nntp;news;
