--- chroot.orig/usr/share/applications/gobby.desktop	2010-02-03 00:00:59.000000000 +0000
+++ chroot/usr/share/applications/gobby.desktop	2011-12-02 12:32:47.124665339 +0000
@@ -9,7 +9,7 @@
 Comment[ja]=共同でテキストファイルを編集する
 Comment[ru]=Редактируйте текстовые файлы совместно
 Comment[uk]=Редагуйте текстові файли разом з іншими користувачами
-Exec=gobby
+Exec=torify gobby
 Terminal=false
 Type=Application
 Icon=gobby.png
