diff --git a/lib/Tails/Persistence/Configuration/Presets.pm b/lib/Tails/Persistence/Configuration/Presets.pm
index 1a01326..9ff213b 100644
--- a/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
+++ b/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
@@ -55,84 +55,74 @@ method _build__presets {
             icon_name   => 'stock_folder',
         },
         {
-            name        => $self->encoding->decode(gettext(q{GnuPG})),
-            description => $self->encoding->decode(gettext(
-                q{GnuPG keyrings and configuration}
-            )),
-            destination => '/home/amnesia/.gnupg',
-            options     => [ 'source=gnupg' ],
-            enabled     => 0,
-            icon_name   => 'seahorse-key',
-        },
-        {
-            name        => $self->encoding->decode(gettext(q{SSH Client})),
+            name        => $self->encoding->decode(gettext(q{Browser bookmarks})),
             description => $self->encoding->decode(gettext(
-                q{SSH keys, configuration and known hosts}
+                q{Bookmarks saved in the Tor Browser}
             )),
-            destination => '/home/amnesia/.ssh',
-            options     => [ 'source=openssh-client'],
+            destination => '/home/amnesia/.mozilla/firefox/bookmarks',
+            options     => [ 'source=bookmarks' ],
             enabled     => 0,
-            icon_name   => 'seahorse-key-ssh',
+            icon_name   => 'user-bookmarks',
         },
         {
-            name        => $self->encoding->decode(gettext(q{Pidgin})),
+            name        => $self->encoding->decode(gettext(q{Network Connections})),
             description => $self->encoding->decode(gettext(
-                q{Pidgin profiles and OTR keyring}
+                q{Configuration of network devices and connections}
             )),
-            destination => '/home/amnesia/.purple',
-            options     => [ 'source=pidgin' ],
+            destination => '/etc/NetworkManager/system-connections',
+            options     => [ 'source=nm-system-connections' ],
             enabled     => 0,
-            icon_name   => 'pidgin',
+            icon_name   => 'network-wired',
         },
         {
-            name        => $self->encoding->decode(gettext(q{Thunderbird})),
+            name        => $self->encoding->decode(gettext(q{APT Packages})),
             description => $self->encoding->decode(gettext(
-                q{Thunderbird profiles and locally stored email}
+                q{Packages downloaded by APT}
             )),
-            destination => '/home/amnesia/.thunderbird',
-            options     => [ 'source=thunderbird' ],
+            destination => '/var/cache/apt/archives',
+            options     => [ 'source=apt/cache' ],
             enabled     => 0,
-            icon_name   => 'thunderbird',
+            icon_name   => 'package-x-generic',
         },
         {
-            name        => $self->encoding->decode(gettext(q{GNOME Keyring})),
+            name        => $self->encoding->decode(gettext(q{APT Lists})),
             description => $self->encoding->decode(gettext(
-                q{Secrets stored by GNOME Keyring}
+                q{Lists downloaded by APT}
             )),
-            destination => '/home/amnesia/.gnome2/keyrings',
-            options     => [ 'source=gnome-keyrings' ],
+            destination => '/var/lib/apt/lists',
+            options     => [ 'source=apt/lists' ],
             enabled     => 0,
-            icon_name   => 'seahorse-key-personal',
+            icon_name   => 'package-x-generic',
         },
         {
-            name        => $self->encoding->decode(gettext(q{Network Connections})),
+            name        => $self->encoding->decode(gettext(q{Printers})),
             description => $self->encoding->decode(gettext(
-                q{Configuration of network devices and connections}
+                q{Printers configuration}
             )),
-            destination => '/etc/NetworkManager/system-connections',
-            options     => [ 'source=nm-system-connections' ],
+            destination => '/etc/cups',
+            options     => [ 'source=cups-configuration' ],
             enabled     => 0,
-            icon_name   => 'network-wired',
+            icon_name   => 'printer',
         },
         {
-            name        => $self->encoding->decode(gettext(q{Browser bookmarks})),
+            name        => $self->encoding->decode(gettext(q{Thunderbird})),
             description => $self->encoding->decode(gettext(
-                q{Bookmarks saved in the Tor Browser}
+                q{Thunderbird profiles and locally stored email}
             )),
-            destination => '/home/amnesia/.mozilla/firefox/bookmarks',
-            options     => [ 'source=bookmarks' ],
+            destination => '/home/amnesia/.thunderbird',
+            options     => [ 'source=thunderbird' ],
             enabled     => 0,
-            icon_name   => 'user-bookmarks',
+            icon_name   => 'thunderbird',
         },
         {
-            name        => $self->encoding->decode(gettext(q{Printers})),
+            name        => $self->encoding->decode(gettext(q{GnuPG})),
             description => $self->encoding->decode(gettext(
-                q{Printers configuration}
+                q{GnuPG keyrings and configuration}
             )),
-            destination => '/etc/cups',
-            options     => [ 'source=cups-configuration' ],
+            destination => '/home/amnesia/.gnupg',
+            options     => [ 'source=gnupg' ],
             enabled     => 0,
-            icon_name   => 'printer',
+            icon_name   => 'seahorse-key',
         },
         {
             name        => $self->encoding->decode(gettext(q{Bitcoin client})),
@@ -145,24 +135,24 @@ method _build__presets {
             icon_name   => 'electrum',
         },
         {
-            name        => $self->encoding->decode(gettext(q{APT Packages})),
+            name        => $self->encoding->decode(gettext(q{Pidgin})),
             description => $self->encoding->decode(gettext(
-                q{Packages downloaded by APT}
+                q{Pidgin profiles and OTR keyring}
             )),
-            destination => '/var/cache/apt/archives',
-            options     => [ 'source=apt/cache' ],
+            destination => '/home/amnesia/.purple',
+            options     => [ 'source=pidgin' ],
             enabled     => 0,
-            icon_name   => 'package-x-generic',
+            icon_name   => 'pidgin',
         },
         {
-            name        => $self->encoding->decode(gettext(q{APT Lists})),
+            name        => $self->encoding->decode(gettext(q{SSH Client})),
             description => $self->encoding->decode(gettext(
-                q{Lists downloaded by APT}
+                q{SSH keys, configuration and known hosts}
             )),
-            destination => '/var/lib/apt/lists',
-            options     => [ 'source=apt/lists' ],
+            destination => '/home/amnesia/.ssh',
+            options     => [ 'source=openssh-client'],
             enabled     => 0,
-            icon_name   => 'package-x-generic',
+            icon_name   => 'seahorse-key-ssh',
         },
         {
             name        => $self->encoding->decode(gettext(q{Dotfiles})),
