[0m[01;34mTails[0m/
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm b/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
index 1a01326..a6bd4b4 100644
--- a/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
+++ b/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
@@ -87,7 +87,7 @@ method _build__presets {
         {
             name        => $self->encoding->decode(gettext(q{Thunderbird})),
             description => $self->encoding->decode(gettext(
-                q{Thunderbird profiles and locally stored email}
+                q{Thunderbird emails, feeds, and settings}
             )),
             destination => '/home/amnesia/.thunderbird',
             options     => [ 'source=thunderbird' ],
