diff --git a/usr/share/perl5/Tails/Persistence.pm b/usr/share/perl5/Tails/Persistence.pm
index 8d1e412..0f1ec7c 100644
--- a/usr/share/perl5/Tails/Persistence.pm
+++ b/usr/share/perl5/Tails/Persistence.pm
@@ -4,10 +4,7 @@ Tails::Persistence - main placeholder module
 
 =cut
 
-use strict;
-use warnings FATAL => 'all';
-use 5.10.0;
-
-
+use 5.10.1;
+use strictures 2;
 
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration.pm b/usr/share/perl5/Tails/Persistence/Configuration.pm
index 9719481..d8edc60 100644
--- a/usr/share/perl5/Tails/Persistence/Configuration.pm
+++ b/usr/share/perl5/Tails/Persistence/Configuration.pm
@@ -5,81 +5,112 @@ Tails::Persistence::Configuration - manage live-persistence.conf and presets
 =cut
 
 package Tails::Persistence::Configuration;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Types::Path::Class;
-use MooseX::Has::Sugar::Saccharin;
-
-
+use Moo;
+use MooX::late;
+use MooX::HandlesVia;
 
+use 5.10.1;
+use strictures 2;
 use autodie qw(:all);
-use warnings FATAL => 'all';
 use Carp;
+use Carp::Assert::More;
 
+use Function::Parameters;
 use Tails::Persistence::Configuration::Atom;
-use Tails::Persistence::Configuration::File;
+use Tails::Persistence::Configuration::ConfigFile;
 use Tails::Persistence::Configuration::Presets;
+use Types::Path::Tiny qw{AbsPath};
 
+use List::MoreUtils qw{none};
 use Locale::gettext;
 use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
+use namespace::clean;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'config_file_path' => lazy_build rw 'Path::Class::File';
-has 'file' =>
-    lazy_build rw 'Tails::Persistence::Configuration::File';
-has 'presets' =>
-    lazy_build rw 'Tails::Persistence::Configuration::Presets';
-has 'atoms' =>
-    lazy_build rw 'ArrayRef[Tails::Persistence::Configuration::Atom]',
-    traits  => [ 'Array' ],
-    handles => {
+has 'config_file_path' => (
+    required  => 1,
+    isa       => AbsPath,
+    is        => 'rw',
+    coerce    => AbsPath->coercion,
+);
+
+has 'file' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Tails::Persistence::Configuration::ConfigFile',
+);
+has 'presets' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Tails::Persistence::Configuration::Presets',
+);
+has 'force_enable_presets' => (
+    is      => 'ro',
+    isa     => 'ArrayRef[Str]',
+    default => sub { [] },
+);
+has 'atoms' => (
+    lazy_build => 1,
+    is            => 'rw',
+    isa           => 'ArrayRef[Tails::Persistence::Configuration::Atom]',
+    handles_via   => 'Array',
+    handles       => {
         all_atoms => 'elements',
-        push_atom  => 'push',
-    };
+        push_atom => 'push',
+    },
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-method _build_file {
-    Tails::Persistence::Configuration::File->new(
-        path => $self->config_file_path
+method _build_file () {
+    my $file = Tails::Persistence::Configuration::ConfigFile->new(
+        config_file_path => $self->config_file_path
     );
+    return $file;
 }
 
-method _build_presets {
+method _build_presets () {
     Tails::Persistence::Configuration::Presets->new();
 }
 
-method _build_atoms {
-    return $self->merge_file_with_presets($self->file, $self->presets);
+method _build_atoms () {
+    return $self->merge_file_with_presets();
 }
 
 =head1 METHODS
 
 =cut
 
-method lines_not_in_presets {
+method lines_not_in_presets () {
     grep {
         my $line = $_;
-        ! grep { $_->equals_line($line) } $self->presets->all
+        ! grep { $_->equals_line($line) } $self->presets->atoms
     } $self->file->all_lines;
 }
 
-method merge_file_with_presets {
-    # Modifying and returning clones of the presets atoms would be a bit cleaner.
+method atoms_not_in_presets () {
+    grep {
+        my $atom = $_;
+        none { $atom->equals_atom($_) } $self->presets->atoms
+    } $self->all_atoms;
+}
+
+method merge_file_with_presets () {
     $self->presets->set_state_from_lines($self->file->all_lines);
+    $self->presets->set_state_from_overrides($self->force_enable_presets);
 
     [
-        $self->presets->all,
+        $self->presets->atoms,
         map {
             Tails::Persistence::Configuration::Atom->new_from_line(
                 $_,
@@ -89,11 +120,11 @@ method merge_file_with_presets {
     ];
 }
 
-method all_enabled_atoms {
+method all_enabled_atoms () {
     grep { $_->enabled } $self->all_atoms;
 }
 
-method all_enabled_lines {
+method all_enabled_lines () {
     map {
         Tails::Persistence::Configuration::Line->new(
             destination => $_->destination,
@@ -102,10 +133,10 @@ method all_enabled_lines {
     } $self->all_enabled_atoms;
 }
 
-method save {
+method save () {
     $self->file->lines([ $self->all_enabled_lines ]);
     $self->file->save;
 }
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/Atom.pm b/usr/share/perl5/Tails/Persistence/Configuration/Atom.pm
index 768b2c5..3433f70 100644
--- a/usr/share/perl5/Tails/Persistence/Configuration/Atom.pm
+++ b/usr/share/perl5/Tails/Persistence/Configuration/Atom.pm
@@ -5,46 +5,49 @@ Tails::Persistence::Configuration::Atom - a GUI-friendly configuration line
 =cut
 
 package Tails::Persistence::Configuration::Atom;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
-
-
+use 5.10.1;
+use strictures 2;
+use Moo;
+use MooX::late;
 
 with 'Tails::Persistence::Role::ConfigurationLine';
-with 'Tails::Role::HasEncoding';
 
 use autodie qw(:all);
 use warnings FATAL => 'all';
 
+use Function::Parameters;
 use List::MoreUtils qw{all pairwise};
+use Types::Standard qw(ClassName InstanceOf);
 
-use Locale::gettext;
-use POSIX;
-setlocale(LC_MESSAGES, "");
-textdomain("tails-persistence-setup");
+use namespace::clean;
 
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'enabled'     => required rw Bool;
-has 'name'        => lazy_build rw Str;
-has 'description' => lazy_build rw Str;
-has 'icon_name'     => lazy_build rw Str;
+has 'enabled' => (
+    required => 1,
+    is       => 'rw',
+    isa      => 'Bool',
+);
+
+has 'id' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Str',
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-sub new_from_line {
-    my $class = shift;
-    my $line  = shift;
-    my %args  = @_;
-
+method new_from_line (
+    ClassName $class:
+    (InstanceOf['Tails::Persistence::Configuration::Line']) $line,
+    %args
+) {
     Tails::Persistence::Configuration::Atom->new(
         destination => $line->destination,
         options     => $line->options,
@@ -52,30 +55,33 @@ sub new_from_line {
     );
 }
 
-method _build_name { $self->encoding->decode(gettext('Custom')); }
-
-method _build_description {
-    $self->encoding->decode($self->destination);
+method _build_id () {
+    'Custom';
 }
 
-method _build_icon_name { 'dialog-question' }
-
 
 =head1 METHODS
 
 =cut
 
+method equals_atom (
+    (InstanceOf['Tails::Persistence::Configuration::Atom']) $other_atom
+) {
+    $self->destination eq $other_atom->destination
+        and
+    $self->options_are(@{$other_atom->options});
+}
 
-method equals_line ($line) {
+method equals_line (
+    (InstanceOf['Tails::Persistence::Configuration::Line']) $line
+) {
     $self->destination eq $line->destination
         and
     $self->options_are($line->all_options);
 }
 
-sub options_are {
-    my $self = shift;
-
-    my @expected = sort(@_);
+method options_are (@expected_options) {
+    my @expected = sort(@expected_options);
     my @options  = sort($self->all_options);
 
     return unless @expected == @options;
@@ -85,5 +91,5 @@ sub options_are {
     } @expected, @options;
 }
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/Button.pm b/usr/share/perl5/Tails/Persistence/Configuration/Button.pm
deleted file mode 100644
index 3540544..0000000
--- a/usr/share/perl5/Tails/Persistence/Configuration/Button.pm
+++ /dev/null
@@ -1,163 +0,0 @@
-=head1 NAME
-
-Tails::Persistence::Configuration::Button - a configuration button in the GUI
-
-=cut
-
-package Tails::Persistence::Configuration::Button;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
-
-
-
-use autodie qw(:all);
-use warnings FATAL => 'all';
-use Glib qw{TRUE FALSE};
-use Pango;
-
-
-=head1 ATTRIBUTES
-
-=cut
-
-has 'atom'         => lazy_build ro 'Tails::Persistence::Configuration::Atom',
-    handles => [
-        qw{source destination all_options enabled name description icon_name}
-    ];
-has 'icon'         => lazy_build ro 'Gtk3::Image';
-has 'checked_img'  => lazy_build ro 'Gtk3::Image';
-has 'main_widget'  => lazy_build ro 'Gtk3::ToggleButton',
-    handles => {
-        is_active  => 'get_active',
-        set_active => 'set_active',
-    };
-has 'icon_theme' => ro 'Gtk3::IconTheme', builder '_build_icon_theme';
-has 'title_label'  => lazy_build ro 'Gtk3::Label';
-has 'description_label' => lazy_build ro 'Gtk3::Label';
-
-
-=head1 CONSTRUCTORS
-
-=cut
-
-method _build_icon {
-    Gtk3::Image->new_from_pixbuf($self->icon_theme->load_icon(
-        $self->icon_name, 48, 'use-builtin'
-    ));
-}
-
-method _build_checked_img {
-    # FIXME: we want the GTK_ICON_SIZE_SMALL_TOOLBAR
-    # member of the GtkIconSize enum, not a magic "16" number here.
-    Gtk3::Image->new_from_stock('gtk-apply', 16);
-}
-
-method _build_main_widget {
-    my $line = shift;
-
-    my $vbox = Gtk3::VBox->new();
-    $vbox->set_border_width(5);
-    $vbox->pack_start($self->title_label, FALSE, FALSE, 0);
-    $vbox->pack_start($self->description_label, FALSE, FALSE, 0);
-
-    my $hbox = Gtk3::HBox->new();
-    $hbox->pack_start($self->icon, FALSE, FALSE, 0);
-    $hbox->pack_start($vbox, TRUE, TRUE, 0);
-    $hbox->pack_start($self->checked_img, FALSE, FALSE, 0);
-
-    my $button = Gtk3::ToggleButton->new();
-    $button->add($hbox);
-    $button->set_active($self->enabled);
-    $button->signal_connect('toggled' => sub { $self->toggled_cb });
-
-    $button->signal_connect('show' => sub { $self->refresh_display });
-
-    return $button;
-}
-
-method _build_icon_theme {
-    my $theme = Gtk3::IconTheme::get_default();
-
-    $theme->append_search_path('/usr/share/pixmaps/cryptui/48x48');
-    $theme->append_search_path('/usr/share/pixmaps/seahorse/48x48');
-    $theme->append_search_path('/usr/share/icons/gnome-colors-common/32x32/apps/');
-    $theme->append_search_path('/usr/share/app-install/icons/');
-
-    return $theme;
- }
-
-method _build_title_label {
-    my $title = Gtk3::Label->new($self->name);
-    $title->set_alignment(0.0, 0.5);
-    my $title_attrlist  = Pango::AttrList->new;
-    $title_attrlist->insert($_)
-        foreach ( Pango::AttrScale->new(1.1),Pango::AttrWeight->new('bold') );
-    $title->set_attributes($title_attrlist);
-
-    return $title;
-}
-
-method _build_description_label {
-    my $description = Gtk3::Label->new($self->description);
-    $description->set_alignment(0.0, 0.5);
-    my $description_attrlist = Pango::AttrList->new;
-    $description_attrlist->insert(
-        Pango::AttrForeground->new(30000, 30000, 30000)
-      );
-    $description->set_attributes($description_attrlist);
-    $description->set_line_wrap(TRUE);
-    $description->set_line_wrap_mode('word');
-    $description->set_single_line_mode(FALSE);
-
-    return $description;
-}
-
-
-=head1 METHODS
-
-=cut
-
-method toggled_cb {
-    $self->atom->enabled($self->is_active);
-    $self->refresh_display;
-}
-
-method refresh_display {
-    $self->refresh_checked_img;
-    $self->refresh_text_color;
-}
-
-method refresh_checked_img {
-    if ($self->enabled) {
-        $self->checked_img->show;
-    }
-    else {
-        $self->checked_img->hide;
-    }
-}
-
-method refresh_text_color {
-    my %color = $self->enabled ?
-        (
-            title       => 0,
-            description => 30000,
-    ) : (
-            title       => 30000,
-            description => 40000,
-    );
-    my $attrlist = $self->title_label->get_attributes;
-    $attrlist->change(
-        Pango::AttrForeground->new($color{title}, $color{title}, $color{title})
-    );
-    $self->title_label->set_attributes($attrlist);
-    $attrlist = $self->description_label->get_attributes;
-    $attrlist->change(
-        Pango::AttrForeground->new($color{description}, $color{description}, $color{description})
-    );
-    $self->description_label->set_attributes($attrlist);
-}
-
-no Moose;
-1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/ConfigFile.pm b/usr/share/perl5/Tails/Persistence/Configuration/ConfigFile.pm
new file mode 100644
index 0000000..d5ac42d
--- /dev/null
+++ b/usr/share/perl5/Tails/Persistence/Configuration/ConfigFile.pm
@@ -0,0 +1,96 @@
+=head1 NAME
+
+Tails::Persistence::Configuration::ConfigFile - read, parse and write live-persistence.conf
+
+=cut
+
+package Tails::Persistence::Configuration::ConfigFile;
+use 5.10.1;
+use strictures 2;
+use Moo;
+use MooX::late;
+use MooX::HandlesVia;
+
+use autodie qw(:all);
+use warnings FATAL => 'all';
+use Carp;
+use Function::Parameters;
+use Tails::Persistence::Configuration::Line;
+use Types::Path::Tiny qw{AbsPath};
+
+use namespace::clean;
+
+
+=head1 ATTRIBUTES
+
+=cut
+
+has 'config_file_path' => (
+    isa      => AbsPath,
+    is       => 'ro',
+    coerce   => AbsPath->coercion,
+    required => 1,
+);
+
+has 'lines' => (
+    lazy_build    => 1,
+    is            => 'rw',
+    isa           => 'ArrayRef[Tails::Persistence::Configuration::Line]',
+    handles_via   => [ 'Array' ],
+    handles       => {
+        all_lines => 'elements',
+    },
+);
+
+
+=head1 CONSTRUCTORS
+
+=cut
+
+method BUILD (@args) {
+    $self->config_file_path->touch;
+}
+
+method _build_lines () {
+    return [
+        grep { defined $_ } map {
+            Tails::Persistence::Configuration::Line->new_from_string($_)
+        } $self->config_file_path->lines({chomp => 1})
+    ];
+}
+
+
+=head1 METHODS
+
+=cut
+
+=head2 output
+
+Returns the in-memory configuration, as a string in the live-persistence.conf format.
+
+=cut
+method output () {
+    my $out = "";
+    foreach ($self->all_lines) {
+        $out .= $_->stringify . "\n";
+    }
+    return $out;
+}
+
+=head2 save
+
+Save the in-memory configuration to disk.
+Throw exception on error.
+
+=cut
+method save () {
+    my $output = $self->output;
+    my $fh = $self->config_file_path->openw();
+    print $fh $output;
+    $fh->sync;
+    close $fh;
+    $self->config_file_path->chmod(0600);
+}
+
+no Moo;
+1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/File.pm b/usr/share/perl5/Tails/Persistence/Configuration/File.pm
deleted file mode 100644
index aa76c16..0000000
--- a/usr/share/perl5/Tails/Persistence/Configuration/File.pm
+++ /dev/null
@@ -1,94 +0,0 @@
-=head1 NAME
-
-Tails::Persistence::Configuration::File - read, parse and write live-persistence.conf
-
-=cut
-
-package Tails::Persistence::Configuration::File;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Types::Path::Class;
-use MooseX::Has::Sugar::Saccharin;
-
-
-
-use autodie qw(:all);
-use warnings FATAL => 'all';
-use Carp;
-use Path::Class;
-use Tails::Persistence::Configuration::Line;
-
-
-=head1 ATTRIBUTES
-
-=cut
-
-has 'path' => (
-    isa        => 'Path::Class::File',
-    is         => 'ro',
-    coerce     => 1,
-    required   => 1,
-);
-
-has 'lines' =>
-    lazy_build rw 'ArrayRef[Tails::Persistence::Configuration::Line]',
-    traits  => [ 'Array' ],
-    handles => {
-        all_lines => 'elements',
-    };
-
-
-=head1 CONSTRUCTORS
-
-=cut
-
-sub BUILD {
-    my $self = shift;
-
-    $self->path->touch;
-}
-
-method _build_lines {
-    return [
-        grep { defined $_ } map {
-            Tails::Persistence::Configuration::Line->new_from_string($_)
-        } $self->path->slurp(chomp => 1)
-    ];
-}
-
-
-=head1 METHODS
-
-=cut
-
-=head2 output
-
-Returns the in-memory configuration, as a string in the live-persistence.conf format.
-
-=cut
-method output {
-    my $out = "";
-    foreach ($self->all_lines) {
-        $out .= $_->stringify . "\n";
-    }
-    return $out;
-}
-
-=head2 save
-
-Save the in-memory configuration to disk.
-Throw exception on error.
-
-=cut
-method save {
-    my $output = $self->output;
-    my $fh = $self->path->openw();
-    print $fh $output;
-    $fh->sync;
-    close $fh;
-    chmod 0600, $self->path;
-}
-
-no Moose;
-1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/Line.pm b/usr/share/perl5/Tails/Persistence/Configuration/Line.pm
index f3fd304..a21e9f8 100644
--- a/usr/share/perl5/Tails/Persistence/Configuration/Line.pm
+++ b/usr/share/perl5/Tails/Persistence/Configuration/Line.pm
@@ -5,11 +5,12 @@ Tails::Persistence::Configuration::Line - read, parse and write live-persistence
 =cut
 
 package Tails::Persistence::Configuration::Line;
-
-use Moose;
+use 5.10.1;
+use strictures 2;
+use Moo;
 with 'Tails::Persistence::Role::ConfigurationLine';
 
+use namespace::clean;
 
-
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm b/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
index 1a01326..befc2f0 100644
--- a/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
+++ b/usr/share/perl5/Tails/Persistence/Configuration/Presets.pm
@@ -5,180 +5,248 @@ Tails::Persistence::Configuration::Presets - available configuration snippets
 =cut
 
 package Tails::Persistence::Configuration::Presets;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
-
-
+use 5.10.1;
+use strictures 2;
+use Moo;
+use MooX::HandlesVia;
+use MooX::late;
 
 with 'Tails::Role::HasEncoding';
 
 use autodie qw(:all);
 use warnings FATAL => 'all';
 use Carp;
+use Function::Parameters;
+use List::MoreUtils qw{all};
 use Tails::Persistence::Configuration::Atom;
+use Types::Standard qw(ArrayRef Str);
 
 use Locale::gettext;
 use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
+use namespace::clean;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
-has '_presets' =>
-    lazy_build ro 'ArrayRef[Tails::Persistence::Configuration::Atom]',
-    traits  => [ 'Array' ],
-    handles => {
-        count => 'count',
-        all   => 'elements',
-    };
+has '_presets' => (
+    lazy_build  => 1,
+    is          => 'rw',
+    isa         => 'ArrayRef',
+    handles_via => 'Array',
+    handles     => {
+        count   => 'count',
+        all     => 'elements',
+    },
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-method _build__presets {
+method _build__presets () {
     my @presets = (
         {
+            id          => 'PersonalData',
             name        => $self->encoding->decode(gettext(q{Personal Data})),
             description => $self->encoding->decode(gettext(
                 q{Keep files stored in the `Persistent' directory}
             )),
-            destination => '/home/amnesia/Persistent',
-            enabled     => 1,
-            options     => [ 'source=Persistent' ],
             icon_name   => 'stock_folder',
+            enabled     => 1,
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/Persistent',
+                    options     => [ 'source=Persistent' ],
+                },
+            ]
         },
         {
-            name        => $self->encoding->decode(gettext(q{GnuPG})),
-            description => $self->encoding->decode(gettext(
-                q{GnuPG keyrings and configuration}
-            )),
-            destination => '/home/amnesia/.gnupg',
-            options     => [ 'source=gnupg' ],
-            enabled     => 0,
-            icon_name   => 'seahorse-key',
-        },
-        {
-            name        => $self->encoding->decode(gettext(q{SSH Client})),
-            description => $self->encoding->decode(gettext(
-                q{SSH keys, configuration and known hosts}
-            )),
-            destination => '/home/amnesia/.ssh',
-            options     => [ 'source=openssh-client'],
-            enabled     => 0,
-            icon_name   => 'seahorse-key-ssh',
-        },
-        {
-            name        => $self->encoding->decode(gettext(q{Pidgin})),
+            id          => 'BrowserBookmarks',
+            name        => $self->encoding->decode(gettext(q{Browser Bookmarks})),
             description => $self->encoding->decode(gettext(
-                q{Pidgin profiles and OTR keyring}
+                q{Bookmarks saved in the Tor Browser}
             )),
-            destination => '/home/amnesia/.purple',
-            options     => [ 'source=pidgin' ],
+            icon_name   => 'user-bookmarks',
             enabled     => 0,
-            icon_name   => 'pidgin',
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/.mozilla/firefox/bookmarks',
+                    options     => [ 'source=bookmarks' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{Thunderbird})),
+            id          => 'NetworkConnections',
+            name        => $self->encoding->decode(gettext(q{Network Connections})),
             description => $self->encoding->decode(gettext(
-                q{Thunderbird profiles and locally stored email}
+                q{Configuration of network devices and connections}
             )),
-            destination => '/home/amnesia/.thunderbird',
-            options     => [ 'source=thunderbird' ],
+            icon_name   => 'network-wired',
             enabled     => 0,
-            icon_name   => 'thunderbird',
+            atoms_args  => [
+                {
+                    destination => '/etc/NetworkManager/system-connections',
+                    options     => [ 'source=nm-system-connections' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{GNOME Keyring})),
-            description => $self->encoding->decode(gettext(
-                q{Secrets stored by GNOME Keyring}
+            id                  => 'AdditionalSoftware',
+            name                => $self->encoding->decode(gettext(q{Additional Software})),
+            description         => $self->encoding->decode(gettext(
+                q{Software installed when starting Tails}
             )),
-            destination => '/home/amnesia/.gnome2/keyrings',
-            options     => [ 'source=gnome-keyrings' ],
-            enabled     => 0,
-            icon_name   => 'seahorse-key-personal',
+            icon_name           => 'package-x-generic',
+            enabled             => 0,
+            configuration_app_desktop_id => 'org.boum.tails.additional-software-config.desktop',
+            atoms_args          => [
+                {
+                    destination => '/var/cache/apt/archives',
+                    options     => [ 'source=apt/cache' ],
+                },
+                {
+                    destination => '/var/lib/apt/lists',
+                    options     => [ 'source=apt/lists' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{Network Connections})),
+            id          => 'Printers',
+            name        => $self->encoding->decode(gettext(q{Printers})),
             description => $self->encoding->decode(gettext(
-                q{Configuration of network devices and connections}
+                q{Printers configuration}
             )),
-            destination => '/etc/NetworkManager/system-connections',
-            options     => [ 'source=nm-system-connections' ],
+            icon_name   => 'printer',
             enabled     => 0,
-            icon_name   => 'network-wired',
+            atoms_args  => [
+                {
+                    destination => '/etc/cups',
+                    options     => [ 'source=cups-configuration' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{Browser bookmarks})),
+            id          => 'Thunderbird',
+            name        => $self->encoding->decode(gettext(q{Thunderbird})),
             description => $self->encoding->decode(gettext(
-                q{Bookmarks saved in the Tor Browser}
+                q{Thunderbird emails, feeds, and settings}
             )),
-            destination => '/home/amnesia/.mozilla/firefox/bookmarks',
-            options     => [ 'source=bookmarks' ],
+            icon_name   => 'thunderbird',
             enabled     => 0,
-            icon_name   => 'user-bookmarks',
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/.thunderbird',
+                    options     => [ 'source=thunderbird' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{Printers})),
+            id          => 'GnuPG',
+            name        => $self->encoding->decode(gettext(q{GnuPG})),
             description => $self->encoding->decode(gettext(
-                q{Printers configuration}
+                q{GnuPG keyrings and configuration}
             )),
-            destination => '/etc/cups',
-            options     => [ 'source=cups-configuration' ],
+            icon_name   => 'seahorse-key',
             enabled     => 0,
-            icon_name   => 'printer',
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/.gnupg',
+                    options     => [ 'source=gnupg' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{Bitcoin client})),
+            id          => 'BitcoinClient',
+            name        => $self->encoding->decode(gettext(q{Bitcoin Client})),
             description => $self->encoding->decode(gettext(
                 q{Electrum's bitcoin wallet and configuration}
             )),
-            destination => '/home/amnesia/.electrum',
-            options     => [ 'source=electrum' ],
-            enabled     => 0,
             icon_name   => 'electrum',
+            enabled     => 0,
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/.electrum',
+                    options     => [ 'source=electrum' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{APT Packages})),
+            id          => 'Pidgin',
+            name        => $self->encoding->decode(gettext(q{Pidgin})),
             description => $self->encoding->decode(gettext(
-                q{Packages downloaded by APT}
+                q{Pidgin profiles and OTR keyring}
             )),
-            destination => '/var/cache/apt/archives',
-            options     => [ 'source=apt/cache' ],
+            icon_name   => 'pidgin',
             enabled     => 0,
-            icon_name   => 'package-x-generic',
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/.purple',
+                    options     => [ 'source=pidgin' ],
+                },
+            ],
         },
         {
-            name        => $self->encoding->decode(gettext(q{APT Lists})),
+            id          => 'SSHClient',
+            name        => $self->encoding->decode(gettext(q{SSH Client})),
             description => $self->encoding->decode(gettext(
-                q{Lists downloaded by APT}
+                q{SSH keys, configuration and known hosts}
             )),
-            destination => '/var/lib/apt/lists',
-            options     => [ 'source=apt/lists' ],
+            icon_name   => 'seahorse-key-ssh',
             enabled     => 0,
-            icon_name   => 'package-x-generic',
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia/.ssh',
+                    options     => [ 'source=openssh-client'],
+                },
+            ],
         },
         {
+            id          => 'Dotfiles',
             name        => $self->encoding->decode(gettext(q{Dotfiles})),
             description => $self->encoding->decode(gettext(
                 q{Symlink into $HOME every file or directory found in the `dotfiles' directory}
             )),
-            destination => '/home/amnesia',
-            options     => [ 'source=dotfiles', 'link' ],
-            enabled     => 0,
             icon_name   => 'preferences-desktop',
+            enabled     => 0,
+            atoms_args  => [
+                {
+                    destination => '/home/amnesia',
+                    options     => [ 'source=dotfiles', 'link' ],
+                },
+            ],
         },
     );
 
-    return [
-        map { Tails::Persistence::Configuration::Atom->new(%{$_}) } @presets
-    ];
+    foreach my $preset (@presets) {
+        my @atoms;
+        foreach my $atom_init_args (@{$preset->{atoms_args}}) {
+            push @atoms, Tails::Persistence::Configuration::Atom->new(
+                id          => $preset->{id},
+                name        => $preset->{name},
+                description => $preset->{description},
+                enabled     => $preset->{enabled},
+                %{$atom_init_args}
+            );
+        }
+        $preset->{atoms} = \@atoms;
+        delete $preset->{atom_args};
+    }
+
+    return \@presets;
+}
+
+method atoms () {
+    my @atoms;
+    foreach my $preset ($self->all) {
+        push @atoms, @{$preset->{atoms}};
+    }
+    return @atoms;
 }
 
 
@@ -186,14 +254,20 @@ method _build__presets {
 
 =cut
 
-sub set_state_from_lines {
-    my $self = shift;
-    my @lines = @_;
-
-    foreach my $atom ($self->all) {
+method set_state_from_lines (@lines) {
+    foreach my $atom ($self->atoms) {
         $atom->enabled(1) if grep { $atom->equals_line($_) } @lines;
     }
+    foreach my $preset ($self->all) {
+        $self->{enabled} = all { $_->enabled } @{$preset->{atoms}};
+    }
+}
+
+method set_state_from_overrides (ArrayRef[Str] $overrides) {
+    foreach my $atom ($self->atoms) {
+        $atom->enabled(1) if grep { $atom->id eq $_ } @$overrides;
+    }
 }
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Configuration/Setting.pm b/usr/share/perl5/Tails/Persistence/Configuration/Setting.pm
new file mode 100644
index 0000000..5a3df18
--- /dev/null
+++ b/usr/share/perl5/Tails/Persistence/Configuration/Setting.pm
@@ -0,0 +1,242 @@
+=head1 NAME
+
+Tails::Persistence::Configuration::Setting - a persistence feature displayed in the GUI
+
+=cut
+
+package Tails::Persistence::Configuration::Setting;
+
+use 5.10.1;
+use strictures 2;
+use autodie qw(:all);
+use Function::Parameters;
+use Glib qw{TRUE FALSE};
+use List::MoreUtils qw{all};
+use Pango;
+use UUID::Tiny ':std';
+
+use Glib::Object::Introspection;
+Glib::Object::Introspection->setup(
+    basename => 'Gio',
+    version  => '2.0',
+    package  => 'Gio'
+);
+
+use Locale::gettext;
+use POSIX;
+setlocale(LC_MESSAGES, "");
+textdomain("tails-persistence-setup");
+
+use Moo;
+use MooX::late;
+with 'Tails::Role::HasEncoding';
+use namespace::clean;
+
+
+=head1 ATTRIBUTES
+
+=cut
+
+has 'atoms' => (
+    required    => 1,
+    is          => 'ro',
+    isa         => 'ArrayRef[Tails::Persistence::Configuration::Atom]',
+);
+
+foreach (qw{id name description icon_name}) {
+    has $_ => (
+        lazy_build => 1,
+        is         => 'rw',
+        isa        => 'Str',
+    );
+}
+
+has 'icon' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::Image',
+);
+
+has 'main_widget' => (
+    lazy_build     => 1,
+    is             => 'ro',
+    isa            => 'Gtk3::HBox',
+);
+
+has 'switch' => (
+    lazy_build     => 1,
+    is             => 'ro',
+    isa            => 'Gtk3::Switch',
+    handles        => {
+        is_active  => 'get_active',
+        set_active => 'set_active',
+    },
+);
+
+has 'icon_theme' => (
+    is      => 'ro',
+    isa     => 'Gtk3::IconTheme',
+    builder => '_build_icon_theme',
+);
+
+has 'title_label' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::Label',
+);
+
+has 'description_label' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::Label',
+);
+
+has 'configuration_button' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::Button',
+);
+
+has 'configuration_app_desktop_id' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Str',
+    predicate  => 1,
+);
+
+
+=head1 CONSTRUCTORS
+
+=cut
+
+method _build_name () {
+    $self->encoding->decode(gettext('Custom'));
+}
+
+method _build_id () {
+    create_uuid();
+}
+
+method _build_description () {
+    join(', ', map { $self->encoding->decode($_->destination) } @{$self->atoms});
+}
+
+method _build_icon_name () {
+    'dialog-question'
+}
+
+method _build_icon () {
+    Gtk3::Image->new_from_pixbuf($self->icon_theme->load_icon(
+        $self->icon_name, 48, 'use-builtin'
+    ));
+}
+
+method _build_switch () {
+    my $switch = Gtk3::Switch->new();
+    $switch->set_active($self->enabled);
+    $switch->set_valign('GTK_ALIGN_CENTER');
+    $switch->set_vexpand(FALSE);
+    for (qw{start end top bottom}) {
+        my $method = "set_margin_${_}";
+        $switch->$method(12);
+    }
+    $switch->signal_connect('notify::active' => sub { $self->toggled_cb });
+    return $switch;
+}
+
+method _build_configuration_button () {
+    my $button = Gtk3::Button->new_from_icon_name(
+        'emblem-system-symbolic', 1,
+    );
+    $button->set_valign('GTK_ALIGN_CENTER');
+    $button->signal_connect('clicked' => sub { $self->configuration_cb() });
+
+    return $button;
+}
+
+method _build_main_widget () {
+    my $main_box = Gtk3::HBox->new();
+    $main_box->set_border_width(5);
+
+    my $text_box = Gtk3::VBox->new();
+    $text_box->set_border_width(5);
+    $text_box->pack_start($self->title_label, FALSE, FALSE, 0);
+    $text_box->pack_start($self->description_label, FALSE, FALSE, 0);
+    $text_box->set_margin_start(12);
+    $text_box->set_margin_end(12);
+    $text_box->set_margin_top(6);
+    $text_box->set_margin_bottom(6);
+
+    $main_box->pack_start($self->icon, FALSE, FALSE, 0);
+    $main_box->pack_start($text_box, TRUE, TRUE, 0);
+    if ($self->has_configuration_app_desktop_id) {
+        $main_box->pack_start($self->configuration_button, FALSE, FALSE, 0);
+    }
+    $main_box->pack_start($self->switch, FALSE, FALSE, 0);
+
+    return $main_box;
+}
+
+method _build_icon_theme () {
+    my $theme = Gtk3::IconTheme::get_default();
+
+    $theme->append_search_path('/usr/share/pixmaps/cryptui/48x48');
+    $theme->append_search_path('/usr/share/pixmaps/seahorse/48x48');
+    $theme->append_search_path('/usr/share/icons/gnome-colors-common/32x32/apps/');
+    $theme->append_search_path('/usr/share/app-install/icons/');
+
+    return $theme;
+ }
+
+method _build_title_label () {
+    my $title = Gtk3::Label->new($self->name);
+    $title->set_alignment(0.0, 0.5);
+    my $title_attrlist = Pango::AttrList->new;
+    $title_attrlist->insert($_)
+        foreach ( Pango::AttrScale->new(1.1),Pango::AttrWeight->new('bold') );
+    $title->set_attributes($title_attrlist);
+
+    return $title;
+}
+
+method _build_description_label () {
+    my $description = Gtk3::Label->new($self->description);
+    $description->set_alignment(0.0, 0.5);
+    my $description_attrlist = Pango::AttrList->new;
+    $description_attrlist->insert(
+        Pango::AttrForeground->new(30000, 30000, 30000)
+      );
+    $description->set_attributes($description_attrlist);
+    $description->set_line_wrap(TRUE);
+    $description->set_line_wrap_mode('word');
+    $description->set_single_line_mode(FALSE);
+
+    return $description;
+}
+
+
+=head1 METHODS
+
+=cut
+
+method enabled () {
+    all { $_->enabled } @{$self->atoms};
+}
+
+method toggled_cb () {
+    foreach my $atom (@{$self->atoms}) {
+        $atom->enabled($self->is_active)
+    }
+}
+
+method configuration_cb () {
+    my $configuration_desktop_app_info = Gio::DesktopAppInfo->new(
+        $ENV{DEV_MODE}
+            ? 'yelp.desktop'
+            : $self->configuration_app_desktop_id,
+    );
+    $configuration_desktop_app_info->launch();
+}
+
+no Moo;
+1;
diff --git a/usr/share/perl5/Tails/Persistence/Constants.pm b/usr/share/perl5/Tails/Persistence/Constants.pm
index 8bdb541..4957e6e 100644
--- a/usr/share/perl5/Tails/Persistence/Constants.pm
+++ b/usr/share/perl5/Tails/Persistence/Constants.pm
@@ -1,52 +1,77 @@
 package Tails::Persistence::Constants;
-use Moose;
-use MooseX::Has::Sugar::Saccharin;
-use MooseX::Types::Moose qw{:all};
-use MooseX::Types::Path::Class;
+use 5.10.1;
+use strictures 2;
+use Moo;
+use MooX::late;
+use Types::Path::Tiny qw{AbsPath};
 
 use autodie qw(:all);
-use English qw{-no_match_vars};
-use Method::Signatures::Simple;
+use Function::Parameters;
+
+use namespace::clean;
 
 
 =head1 Attributes
 
 =cut
 for (qw{partition_label partition_guid filesystem_type filesystem_label}) {
-    has "persistence_$_" => lazy_build ro Str;
+    has "persistence_$_" => (
+        lazy_build => 1,
+        is         => 'ro',
+        isa        => 'Str',
+    );
 }
 
-has "persistence_minimum_size"       => lazy_build ro Int;
+has "persistence_minimum_size"       => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Int',
+);
 
-has 'persistence_filesystem_options' => lazy_build ro 'HashRef[Str]';
+has 'persistence_filesystem_options' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'HashRef[Str]',
+);
 
-has 'persistence_state_file'         =>
-    isa        => 'Path::Class::File',
+has 'persistence_state_file'         => (
+    isa        => AbsPath,
     is         => 'ro',
     lazy_build => 1,
-    coerce     => 1,
-    documentation => q{File where tails-greeter writes persistence state.};
+    coerce     => AbsPath->coercion,
+    documentation => q{File where tails-greeter writes persistence state.},
+);
 
 
 =head1 Constructors
 
 =cut
-method _build_persistence_partition_label  { 'TailsData'  }
-method _build_persistence_minimum_size     { 64 * 2 ** 20 }
-method _build_persistence_filesystem_type  { 'ext4'       }
-method _build_persistence_filesystem_label { 'TailsData'  }
+method _build_persistence_partition_label () {
+    'TailsData'
+}
+method _build_persistence_minimum_size () {
+    64 * 2 ** 20
+}
+method _build_persistence_filesystem_type () {
+    'ext4'
+}
+method _build_persistence_filesystem_label () {
+    'TailsData'
+}
 
-method _build_persistence_filesystem_options {
+method _build_persistence_filesystem_options () {
     {
         label => $self->persistence_filesystem_label,
     };
 }
 
-method _build_persistence_partition_guid  {
+method _build_persistence_partition_guid () {
     '8DA63339-0007-60C0-C436-083AC8230908' # Linux reserved
 }
 
-method _build_persistence_state_file { '/var/lib/live/config/tails.persistence' }
+method _build_persistence_state_file () {
+    '/var/lib/live/config/tails.persistence'
+}
 
-no Moose;
+no Moo;
 1; # End of Tails::Persistence::Constants
diff --git a/usr/share/perl5/Tails/Persistence/Role/ConfigurationLine.pm b/usr/share/perl5/Tails/Persistence/Role/ConfigurationLine.pm
index 4222c8a..aee18f2 100644
--- a/usr/share/perl5/Tails/Persistence/Role/ConfigurationLine.pm
+++ b/usr/share/perl5/Tails/Persistence/Role/ConfigurationLine.pm
@@ -5,40 +5,51 @@ Tails::Persistence::Role::ConfigurationLine - live-persistence.conf data structu
 =cut
 
 package Tails::Persistence::Role::ConfigurationLine;
-use Moose::Role;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Types::Path::Class;
-use MooseX::Has::Sugar::Saccharin;
-
-
-
+use 5.10.1;
+use strictures 2;
 use autodie qw(:all);
-use warnings FATAL => 'all';
 use Carp;
+use Function::Parameters;
+use Types::Standard qw(ClassName Str);
+
+use Moo::Role; # Moo::Role exports all methods declared after it's "use"'d
+use MooX::late;
+use MooX::HandlesVia;
+
+use namespace::clean;
 
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'destination' => lazy_build ro Str;
-has 'options' =>
-    lazy_build ro 'ArrayRef[Str]',
-    traits    => [ 'Array' ],
-    handles   => {
+has 'destination' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Str',
+);
+
+has 'options' => (
+    lazy_build        => 1,
+    is                => 'ro',
+    isa               => 'ArrayRef[Str]',
+    handles_via       => 'Array',
+    handles           => {
         count_options => 'count',
         all_options   => 'elements',
         join_options  => 'join',
         grep_options  => 'grep',
-    };
+    }
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-method _build_options     { return [] }
+method _build_options () {
+    return []
+}
 
 
 =head2 new_from_string
@@ -51,10 +62,7 @@ output:
   - throws exception on badly formatted line
 
 =cut
-sub new_from_string {
-    my $class = shift;
-    my $line  = shift;
-
+method new_from_string (ClassName $class: Str $line) {
     chomp $line;
 
     # skip pure-whitespace lines
@@ -84,7 +92,7 @@ as a string in the live-persistence.conf format,
 with no trailing newline.
 
 =cut
-method stringify {
+method stringify () {
     my $out = $self->destination;
     if ($self->count_options) {
         $out .= "\t" . $self->join_options(',');
@@ -92,5 +100,5 @@ method stringify {
     return $out;
 }
 
-no Moose;
+no Moo::Role;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Role/HasStatusArea.pm b/usr/share/perl5/Tails/Persistence/Role/HasStatusArea.pm
index ff42a6f..f38acef 100644
--- a/usr/share/perl5/Tails/Persistence/Role/HasStatusArea.pm
+++ b/usr/share/perl5/Tails/Persistence/Role/HasStatusArea.pm
@@ -5,12 +5,14 @@ Tails::Persistence::Role::HasStatusArea - status area interface
 =cut
 
 package Tails::Persistence::Role::HasStatusArea;
-use Moose::Role;
-
-
+use 5.10.1;
+use strictures 2;
+use Moo::Role;
 
 requires 'status_area';
 requires 'working';
 
-no Moose 'Role';
+use namespace::clean;
+
+no Moo::Role;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Role/SetupStep.pm b/usr/share/perl5/Tails/Persistence/Role/SetupStep.pm
index adca738..16a18f8 100644
--- a/usr/share/perl5/Tails/Persistence/Role/SetupStep.pm
+++ b/usr/share/perl5/Tails/Persistence/Role/SetupStep.pm
@@ -5,19 +5,12 @@ Tails::Persistence::Role::SetupStep - role for persistence setup steps
 =cut
 
 package Tails::Persistence::Role::SetupStep;
-use Moose::Role;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
 
-
-
-with 'Tails::Role::HasEncoding';
-with 'Tails::Persistence::Role::HasStatusArea';
-
-use 5.10.0;
-use namespace::autoclean;
+use 5.10.1;
+use strictures 2;
 use autodie qw(:all);
 
+use Function::Parameters;
 use Glib qw{TRUE FALSE};
 use Gtk3 qw{-init};
 
@@ -26,7 +19,14 @@ use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
-requires '_build_main_box';
+use Moo::Role;
+use MooX::late;
+use namespace::clean;
+
+with 'Tails::Role::HasEncoding';
+with 'Tails::Persistence::Role::HasStatusArea';
+
+requires '_build_main_widget';
 requires 'go_button_pressed';
 
 
@@ -34,21 +34,45 @@ requires 'go_button_pressed';
 
 =cut
 
-has 'name' => required ro Str;
+has 'name' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Str',
+);
 
-has 'main_box' => lazy_build rw 'Gtk3::VBox';
+has 'main_widget' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::VBox',
+);
 
 foreach (qw{title subtitle description}) {
-    has $_ => lazy_build rw 'Gtk3::Label';
+    has $_ => (
+        lazy_build => 1,
+        is         => 'rw',
+        isa        => 'Gtk3::Label',
+    );
 }
 
-has 'go_button'   => lazy_build rw 'Gtk3::Button';
+has 'go_button'   => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::Button',
+);
 foreach (qw{go_callback success_callback}) {
-    has $_ => required ro CodeRef;
+    has $_ => (
+        required => 1,
+        is       => 'ro',
+        isa      => 'CodeRef',
+    );
 }
 
 foreach (qw{drive_vendor drive_model}) {
-    has $_ => required ro Str;
+    has $_ => (
+        required => 1,
+        is       => 'ro',
+        isa      => 'Str',
+    );
 }
 
 
@@ -56,9 +80,7 @@ foreach (qw{drive_vendor drive_model}) {
 
 =cut
 
-sub _build_title {
-    my $self = shift;
-
+method _build_title () {
     my $label = Gtk3::Label->new('');
     $label->set_alignment(0.0, 0.5);
     my $attrlist  = Pango::AttrList->new;
@@ -70,9 +92,7 @@ sub _build_title {
     return $label;
 }
 
-sub _build_subtitle {
-    my $self = shift;
-
+method _build_subtitle () {
     my $label = Gtk3::Label->new('');
     $label->set_alignment(0.0, 0.5);
     my $attrlist  = Pango::AttrList->new;
@@ -87,9 +107,7 @@ sub _build_subtitle {
     return $label;
 }
 
-sub _build_description {
-    my $self = shift;
-
+method _build_description () {
     my $label = Gtk3::Label->new('');
     $label->set_alignment(0.0, 0.5);
     $label->set_padding(10, 10);
@@ -100,9 +118,7 @@ sub _build_description {
     return $label;
 }
 
-sub _build_go_button {
-    my $self = shift;
-
+method _build_go_button () {
     my $button = Gtk3::Button->new;
     $button->set_sensitive(FALSE);
     $button->set_can_default(TRUE);
@@ -111,5 +127,5 @@ sub _build_go_button {
     return $button;
 }
 
-no Moose::Role;
+no Moo::Role;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Role/StatusArea.pm b/usr/share/perl5/Tails/Persistence/Role/StatusArea.pm
index e0ad103..7a094df 100644
--- a/usr/share/perl5/Tails/Persistence/Role/StatusArea.pm
+++ b/usr/share/perl5/Tails/Persistence/Role/StatusArea.pm
@@ -5,35 +5,45 @@ Tails::Persistence::Role::StatusArea - role to manage a status area
 =cut
 
 package Tails::Persistence::Role::StatusArea;
-use Moose::Role;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
-
-
-
-use 5.10.0;
-use namespace::autoclean;
+use 5.10.1;
+use strictures 2;
 use autodie qw(:all);
+use Function::Parameters;
 use Glib qw{TRUE FALSE};
 use Gtk3;
 
+use Moo::Role;
+use MooX::late;
+use namespace::clean;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'status_area' => lazy_build rw 'Gtk3::HBox';
-has 'spinner'     => lazy_build rw 'Gtk3::Spinner';
-has 'working'     => lazy_build rw Bool;
+has 'status_area' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::HBox',
+);
+has 'spinner'     => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::Spinner',
+);
+has 'working'     => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Bool',
+
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-sub _build_status_area {
-    my $self = shift;
-
+method _build_status_area () {
     my $box = Gtk3::HBox->new(FALSE, 64);
     $box->set_border_width(10);
     $box->pack_start($self->spinner, FALSE, FALSE, 0);
@@ -41,9 +51,7 @@ sub _build_status_area {
     return $box;
 }
 
-sub _build_spinner {
-    my $self = shift;
-
+method _build_spinner () {
     my $spinner = Gtk3::Spinner->new;
     $spinner->set_size_request(80,80);
     return $spinner;
@@ -71,5 +79,5 @@ after 'working' => sub {
     }
 };
 
-no Moose 'Role';
+no Moo::Role;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Setup.pm b/usr/share/perl5/Tails/Persistence/Setup.pm
index d6c0ba2..0a83c5a 100644
--- a/usr/share/perl5/Tails/Persistence/Setup.pm
+++ b/usr/share/perl5/Tails/Persistence/Setup.pm
@@ -5,41 +5,29 @@ Tails::Persistence::Setup - main application class
 =cut
 
 package Tails::Persistence::Setup;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Types::Path::Class;
-use MooseX::Has::Sugar::Saccharin;
 
-
-
-with 'Tails::Role::DisplayError::Gtk3';
-with 'Tails::Role::HasEncoding';
-with 'Tails::Role::HasDBus::System';
-with 'MooseX::Getopt::Dashes';
-
-use 5.10.0;
-use namespace::autoclean;
+use 5.10.1;
+use strictures 2;
 
 use autodie qw(:all);
 use Carp::Assert::More;
-use Data::Dumper;
 use English qw{-no_match_vars};
+use Function::Parameters;
 use Glib qw{TRUE FALSE};
 use Gtk3 qw{-init};
 use Net::DBus qw(:typing);
 use Net::DBus::Annotation qw(:call);
-use List::Util qw{first max};
+use List::Util qw{first min max};
 use Number::Format qw(:subs);
-use Path::Class;
+use Path::Tiny;
+use Types::Standard qw(Str HashRef);
 use Try::Tiny;
+use Types::Path::Tiny qw{Dir};
 
 use Tails::RunningSystem;
 use Tails::UDisks;
-
 use Tails::Persistence::Configuration;
 use Tails::Persistence::Constants;
-
 use Tails::Persistence::Step::Bootstrap;
 use Tails::Persistence::Step::Configure;
 use Tails::Persistence::Step::Delete;
@@ -50,132 +38,196 @@ use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
+no Moo::sification;
+use Moo;
+use MooX::late;
+use MooX::HandlesVia;
+
+with 'Tails::Role::DisplayError::Gtk3';
+with 'Tails::Role::HasDBus::System';
+with 'Tails::Role::HasEncoding';
+
+use namespace::clean;
+
+use MooX::Options;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'verbose' =>
-    ro Bool,
+option 'verbose' => (
+    is            => 'ro',
+    isa           => 'Bool',
     documentation => q{Get more output.},
     default       => sub {
         exists $ENV{DEBUG} && defined $ENV{DEBUG} && $ENV{DEBUG}
-    };
+    },
+);
 
-has 'force' =>
-    lazy_build ro Bool,
-    documentation => q{Make some sanity checks non-fatal.};
+option 'force' => (
+    lazy_build    => 1,
+    is            => 'ro',
+    isa           => 'Bool',
+    documentation => q{Make some sanity checks non-fatal.},
+);
 
-has 'udisks' =>
-    lazy_build ro 'Tails::UDisks',
-    metaclass => 'NoGetopt',
-    handles   => [
+has 'udisks' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Tails::UDisks',
+    handles    => [
         qw{bytes_array_to_string device_has_partition_with_label
            drive_is_optical drive_is_connected_via_a_supported_interface
            device_partition_with_label get_block_device_property
            get_filesystem_property get_partition_property luks_holder
            mountpoints partitions udisks_service}
-    ];
+    ],
+);
 
-has 'running_system' =>
-    lazy_build ro 'Tails::RunningSystem',
-    metaclass => 'NoGetopt',
-    handles   => [
+has 'running_system' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Tails::RunningSystem',
+    handles    => [
         qw{boot_drive boot_block_device boot_device_file boot_drive_model boot_drive_vendor
            boot_drive_size
            started_from_device_installed_with_tails_installer}
-    ];
+    ],
+);
 
-has 'persistence_constants' =>
-    lazy_build ro 'Tails::Persistence::Constants',
-    metaclass => 'NoGetopt',
-    handles   => [
+has 'persistence_constants' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Tails::Persistence::Constants',
+    handles    => [
         map {
             "persistence_$_"
         } qw{partition_label partition_guid filesystem_type filesystem_label
              minimum_size filesystem_options state_file}
-    ];
+    ],
+);
 
-has 'main_window' =>
-    lazy_build ro 'Gtk3::Window',
-    metaclass  => 'NoGetopt';
+has 'main_window' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::Window',
+);
 
-has "$_" => lazy_build ro Str
-    for (qw{override_liveos_mountpoint override_boot_drive
-            override_system_partition});
+option "$_" => (
+    is         => 'ro',
+    format     => 's',
+    isa        => 'Str',
+    predicate  => 1,
+) for (qw{override_liveos_mountpoint override_boot_drive
+          override_system_partition});
 
-has 'persistence_partition_device_file'=> lazy_build ro Str, metaclass => 'NoGetopt';
-has 'persistence_partition_size'     => lazy_build ro Int,  metaclass => 'NoGetopt';
-has 'persistence_is_enabled'         => lazy_build ro Bool, metaclass => 'NoGetopt';
-has 'persistence_is_read_write'      => lazy_build ro Bool, metaclass => 'NoGetopt';
+has 'persistence_partition_device_file' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Str',
+);
+has 'persistence_partition_size' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Int',
+);
+has 'persistence_is_enabled' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Bool',
+);
 
 has 'persistence_partition_mountpoint' => (
-    isa        => 'Path::Class::Dir',
+    isa        => Dir,
     is         => 'rw',
     lazy_build => 1,
-    coerce     => 1,
-    metaclass  => 'NoGetopt',
+    coerce     => Dir->coercion,
 );
 
 foreach (qw{beginning_of_free_space size_of_free_space}) {
-    has $_ => lazy_build ro Int, metaclass  => 'NoGetopt';
+    has $_ => (
+        lazy_build => 1,
+        is         => 'ro',
+        isa        => 'Int',
+    );
 }
 
-has 'current_step' =>
-    rw Object,
-    predicate 'has_current_step',
-    metaclass  => 'NoGetopt';
+has 'current_step' => (
+    is        => 'rw',
+    isa       => 'Object',
+    predicate => 'has_current_step',
+);
 
-has 'steps' =>
-    lazy_build required ro 'ArrayRef[Str]',
-    traits  => ['Array'],
-    handles => {
-        all_steps       => 'elements',
+option 'steps' => (
+    lazy_build    => 1,
+    required      => 1,
+    repeatable    => 1,
+    format        => 's@',
+    is            => 'rw',
+    isa           => 'ArrayRef[Str]',
+    handles_via   => ['Array'],
+    handles       => {
         number_of_steps => 'count',
-        append_to_steps => 'push',
         shift_steps     => 'shift',
         next_step       => 'first',
-        grep_steps      => 'grep',
     },
-    documentation => q{Specify once per wizard step to run. Supported steps are: bootstrap, configure, delete.};
+    documentation => q{Specify once per wizard step to run. Supported steps are: bootstrap, configure, delete.},
+);
 
-has 'orig_steps' =>
-    rw 'ArrayRef[Str]',
-    traits  => ['Array'],
-    handles => {
-        grep_orig_steps => 'grep',
-    };
+has 'configuration' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Tails::Persistence::Configuration',
+    handles    => { save_configuration => 'save' },
+);
 
-has 'passphrase' => rw Str, documentation => q{Unsupported. Developers only.};
+option 'force_enable_presets' => (
+    is            => 'ro',
+    repeatable    => 1,
+    format        => 's@',
+    isa           => 'ArrayRef[Str]',
+    handles_via   => ['Array'],
+    documentation => q{Specify once per additional preset to forcibly enable.},
+    default       => sub { [] },
+);
 
-has 'configuration' =>
-    lazy_build rw 'Tails::Persistence::Configuration',
-    handles    => { save_configuration => 'save' },
-    metaclass  => 'NoGetopt';
+option 'display_finished_message' => (
+    is            => 'ro',
+    isa           => 'Bool',
+    documentation => q{Display an explanatory message once done.},
+    default       => sub { 1 },
+    negativable   => 1,
+);
 
-has '+codeset'  => ( metaclass => 'NoGetopt' );
-has '+encoding' => ( metaclass => 'NoGetopt' );
+option 'gui' => (
+    is            => 'ro',
+    isa           => 'Bool',
+    documentation => q{Display the GUI. Only a few operations are available under --no-gui.},
+    default       => sub { 1 },
+    negativable   => 1,
+);
 
 
 =head1 CONSTRUCTORS AND BUILDERS
 
 =cut
 
-method BUILD {
-    my @orig_steps = $self->all_steps;
-    $self->orig_steps(\@orig_steps);
+method BUILD (@args) {
+    if (! $self->gui) {
+        assert_is($self->number_of_steps, 1,
+                  "Exactly one step is enabled under --no-gui");
+        assert_is($self->steps->[0], 'configure',
+                  "The requested step is 'configure' under --no-gui");
+    }
 }
 
-sub _build_force {
-    my $self = shift;
-    0;
-}
+method _build_force () { 0; }
 
-sub _build_persistence_constants { my $self = shift; Tails::Persistence::Constants->new(); }
-sub _build_udisks { my $self = shift; Tails::UDisks->new(); }
+method _build_persistence_constants () { Tails::Persistence::Constants->new(); }
+method _build_udisks () { Tails::UDisks->new(); }
 
-sub _build_running_system {
-    my $self = shift;
+method _build_running_system () {
 
     my @args;
     for (qw{liveos_mountpoint boot_drive system_partition}) {
@@ -189,9 +241,7 @@ sub _build_running_system {
     Tails::RunningSystem->new(main_window => $self->main_window, @args);
 }
 
-sub _build_persistence_is_enabled {
-    my $self = shift;
-
+method _build_persistence_is_enabled () {
     -e $self->persistence_state_file || return 0;
     -r $self->persistence_state_file || return 0;
 
@@ -201,21 +251,7 @@ sub _build_persistence_is_enabled {
     defined($value) && $value eq 'true';
 }
 
-sub _build_persistence_is_read_write {
-    my $self = shift;
-
-    -e $self->persistence_state_file || return 0;
-    -r $self->persistence_state_file || return 0;
-
-    my $value = $self->get_variable_from_persistence_state_file(
-        'TAILS_PERSISTENCE_READONLY'
-    );
-    ! (defined($value) && $value eq 'true');
-}
-
-sub _build_steps {
-    my $self = shift;
-
+method _build_steps () {
     if ($self->device_has_persistent_volume) {
         return [ qw{configure} ];
     }
@@ -224,14 +260,13 @@ sub _build_steps {
     }
 }
 
-sub _build_main_window {
-    my $self = shift;
+method _build_main_window () {
     my $win = Gtk3::Window->new('toplevel');
     $win->set_title($self->encoding->decode(gettext('Setup Tails persistent volume')));
 
     $win->set_border_width(10);
 
-    $win->add($self->current_step->main_box) if $self->has_current_step;
+    $win->add($self->current_step->main_widget) if $self->has_current_step;
     $win->signal_connect('destroy' => sub { Gtk3->main_quit; });
     $win->signal_connect('key-press-event' => sub {
         my $twin = shift;
@@ -243,18 +278,14 @@ sub _build_main_window {
     return $win;
 }
 
-sub _build_persistence_partition_mountpoint {
-    my $self = shift;
-
+method _build_persistence_partition_mountpoint () {
     first {
            $_ eq '/live/persistence/TailsData_unlocked'
-        or $_ eq '/media/tails-persistence-setup/TailsData'
+        or $_ eq '/media/'.getpwuid($UID).'/TailsData'
     } $self->mountpoints($self->persistence_partition);
 }
 
-sub _build_beginning_of_free_space {
-    my $self = shift;
-
+method _build_beginning_of_free_space () {
     align_up_at_2MiB(
         max(
             map {
@@ -265,36 +296,36 @@ sub _build_beginning_of_free_space {
     );
 }
 
-sub _build_size_of_free_space {
-    my $self = shift;
-
+method _build_size_of_free_space () {
     align_down_at_2MiB(
         $self->get_block_device_property($self->boot_block_device, 'Size')
       - $self->beginning_of_free_space
   );
 }
 
-sub _build_persistence_partition_device_file {
-    my $self = shift;
-
+method _build_persistence_partition_device_file () {
     return $self->bytes_array_to_string($self->get_block_device_property(
         $self->persistence_partition, 'PreferredDevice'
     ));
 }
 
-sub _build_persistence_partition_size {
-    my $self = shift;
-
+method _build_persistence_partition_size () {
     $self->get_block_device_property($self->persistence_partition, 'Size');
 }
 
-sub _build_configuration {
-    my $self = shift;
-
-    my $config_file_path = file($self->persistence_partition_mountpoint, 'persistence.conf');
+method _build_configuration () {
+    my $config_file_path = path($self->persistence_partition_mountpoint, 'persistence.conf');
     if (-e $config_file_path) {
-        my $expected_uid = getpwnam('tails-persistence-setup');
-        my $expected_gid = getgrnam('tails-persistence-setup');
+        # In Tails, tails-persistence-setup runs as the tails-persistence-setup
+        # user and the configuration file must be owned by
+        # tails-persistence-setup:tails-persistence-setup.
+        # When developing outside of Tails, the configuration file is also owned
+        # by the user:group that runs tails-persistence-setup.
+        # So in all cases, we effectively want the configuration file
+        # to be owned by the user:group that runs tails-persistence-setup.
+        my $expected_uid = getuid();
+        my $expected_gid = getgid();
+        $self->debug("Expected ownership: ${expected_uid}:${expected_gid}");
         try {
             check_config_file_permissions(
                 $config_file_path,
@@ -313,11 +344,13 @@ sub _build_configuration {
                 $self->encoding->decode(gettext(
                     $_,
                 )));
+            exit 4;
         };
     }
 
     Tails::Persistence::Configuration->new(
-        config_file_path => $config_file_path
+        config_file_path     => $config_file_path,
+        force_enable_presets => $self->force_enable_presets,
     );
 }
 
@@ -326,16 +359,11 @@ sub _build_configuration {
 
 =cut
 
-sub debug {
-    my $self = shift;
-    my $mesg = shift;
+method debug (Str $mesg) {
     say STDERR $self->encoding->encode($mesg) if $self->verbose;
 }
 
-sub check_sanity {
-    my $self      = shift;
-    my $step_name = shift;
-
+method check_sanity (Str $step_name) {
     my %step_checks = (
         'bootstrap' => [
             {
@@ -350,7 +378,6 @@ sub check_sanity {
                 method  => 'device_has_enough_free_space',
                 message => $self->encoding->decode(gettext(
                     "Device %s has not enough unallocated space.")),
-                needs_device_arg => 1,
             },
         ],
         'delete' => [
@@ -363,7 +390,7 @@ sub check_sanity {
             {
                 method  => 'persistence_is_enabled',
                 message => $self->encoding->decode(gettext(
-                    "Cannot delete the persistent volume while in use. You should restart Tails without persistence.")),
+                    "Cannot delete the persistent volume on %s while in use. You should restart Tails without persistence.")),
                 must_be_false => 1,
             },
         ],
@@ -374,33 +401,28 @@ sub check_sanity {
                     "Device %s has no persistent volume.")),
                 needs_device_arg => 1,
             },
-        ],
-    );
-
-    if (! $self->grep_orig_steps(sub { $_ eq 'bootstrap' })) {
-        push @{$step_checks{configure}}, (
             {
                 method  => 'persistence_partition_is_unlocked',
                 message => $self->encoding->decode(gettext(
-                    "Persistence volume is not unlocked.")),
+                    "Persistence volume on %s is not unlocked.")),
             },
             {
                 method  => 'persistence_filesystem_is_mounted',
                 message => $self->encoding->decode(gettext(
-                    "Persistence volume is not mounted.")),
+                    "Persistence volume on %s is not mounted.")),
             },
             {
                 method  => 'persistence_filesystem_is_readable',
                 message => $self->encoding->decode(gettext(
-                    "Persistence volume is not readable. Permissions or ownership problems?")),
+                    "Persistence volume on %s is not readable. Permissions or ownership problems?")),
             },
             {
                 method  => 'persistence_filesystem_is_writable',
                 message => $self->encoding->decode(gettext(
-                    "Persistence volume is not writable. Maybe it was mounted read-only?")),
+                    "Persistence volume on %s is not writable.")),
             },
-        );
-    }
+        ],
+    );
 
     my @checks = (
         {
@@ -431,7 +453,7 @@ sub check_sanity {
     }
 
     foreach my $check (@checks) {
-        my $check_method = $self->meta->get_method($check->{method});
+        my $check_method = $self->can($check->{method});
         assert_defined($check_method);
         my $res;
         my @args = ($self);
@@ -441,7 +463,7 @@ sub check_sanity {
         elsif (exists($check->{needs_drive_arg}) && $check->{needs_drive_arg}) {
             push @args, $self->boot_drive;
         }
-        $res = $check_method->execute(@args);
+        $res = $check_method->(@args);
         if (exists($check->{must_be_false}) && $check->{must_be_false}) {
             $res = ! $res;
         }
@@ -450,7 +472,7 @@ sub check_sanity {
                 gettext($check->{message}),
                 $self->boot_device_file));
             if ($self->force && exists($check->{can_be_forced}) && $check->{can_be_forced}) {
-                warn "$message",
+                say STDERR "$message",
                      "... but --force is enabled, ignoring results of this sanity check.";
             }
             else {
@@ -467,37 +489,29 @@ sub check_sanity {
     return 1;
 }
 
-sub run {
-    my $self = shift;
-
+method run () {
     $self->debug("Entering Tails::Persistence::Setup::run");
     $self->debug(sprintf("Working on device %s", $self->boot_device_file));
 
     $self->main_window->set_visible(FALSE);
     $self->goto_next_step;
-    $self->debug("Entering main Gtk3 loop.");
-    Gtk3->main;
+    if ($self->gui) {
+        $self->debug("Entering main Gtk3 loop.");
+        Gtk3->main;
+    }
 }
 
-sub device_has_persistent_volume {
-    my $self   = shift;
-    my $device = shift;
-    $device  ||= $self->boot_block_device;
-
+method device_has_persistent_volume ($device = undef) {
+    $device ||= $self->boot_block_device;
     $self->debug("Entering device_has_persistent_volume");
     return $self->device_has_partition_with_label($device, $self->persistence_partition_label);
 }
 
-sub device_has_enough_free_space {
-    my $self   = shift;
-    my $device = shift;
-
+method device_has_enough_free_space () {
     $self->size_of_free_space >= $self->persistence_minimum_size;
 }
 
-sub persistence_partition {
-    my $self = shift;
-
+method persistence_partition () {
     $self->debug("Entering persistence_partition");
     $self->device_partition_with_label(
         $self->boot_block_device,
@@ -505,9 +519,7 @@ sub persistence_partition {
     );
 }
 
-sub create_persistence_partition {
-    my $self = shift;
-    my $opts = shift;
+method create_persistence_partition (HashRef $opts) {
     $opts->{end_cb}    ||= sub { say STDERR "finished." };
 
     $self->debug("Entering create_persistence_partition");
@@ -533,11 +545,11 @@ sub create_persistence_partition {
     $self->debug("waiting...");
 }
 
-sub create_persistent_encrypted_filesystem {
-    my $self = shift;
-    my $opts = shift;
+method create_persistent_encrypted_filesystem (
+    HashRef $opts,
+    $create_partition_reply
+) {
     $opts->{end_cb}    ||= sub { say STDERR "finished." };
-    my $create_partition_reply = shift;
     my ($created_device, $create_partition_error);
 
     $self->debug("Entering create_persistent_encrypted_filesystem");
@@ -574,9 +586,7 @@ sub create_persistent_encrypted_filesystem {
     $self->debug("waiting...");
 }
 
-sub delete_persistence_partition {
-    my $self = shift;
-    (my $opts = shift) ||= {};
+method delete_persistence_partition (HashRef $opts = {}) {
     $opts->{end_cb}    ||= sub { say STDERR "finished." };
 
     $self->debug(sprintf("Deleting partition %s", $self->persistence_partition_device_file));
@@ -602,10 +612,7 @@ sub delete_persistence_partition {
     $self->debug("waiting...");
 }
 
-sub mount_persistence_partition {
-    my $self = shift;
-    my $opts = shift;
-
+method mount_persistence_partition () {
     $self->debug(sprintf("Mounting partition %s", $self->persistence_partition_device_file));
 
     my $luks_holder = $self->luks_holder($self->persistence_partition);
@@ -616,39 +623,45 @@ sub mount_persistence_partition {
          ->Mount(dbus_call_sync, {});
 }
 
-sub empty_main_window {
-    my $self = shift;
-
+method empty_main_window () {
     my $child = $self->main_window->get_child;
     $self->main_window->remove($child) if defined($child);
 }
 
-sub run_current_step {
-    my $self = shift;
+method run_current_step () {
     my ($width, $height) = $self->main_window->get_size();
 
     $self->debug("Running step " . $self->current_step->name);
 
-    $self->current_step->working(0);
-    $self->empty_main_window;
-    $self->main_window->add($self->current_step->main_box);
-    $self->main_window->set_default($self->current_step->go_button);
-    $self->main_window->show_all;
-    $self->current_step->working(0);
-    $self->main_window->set_visible(TRUE);
-
-    if($self->current_step->name eq 'configure') {
-        $self->main_window->resize($width, $self->main_window->get_screen()->get_height());
-    }
-    else {
-        $self->main_window->resize($width, $height);
+    if ($self->gui) {
+        $self->current_step->working(0);
+        $self->empty_main_window;
+        $self->main_window->add($self->current_step->main_widget);
+        $self->main_window->set_default($self->current_step->go_button);
+        $self->main_window->show_all;
+        $self->current_step->working(0);
+        $self->main_window->set_visible(TRUE);
+
+        if($self->current_step->name eq 'configure') {
+            $self->main_window->resize(
+                $width,
+                max(
+                    900,
+                    min(450, $self->main_window->get_screen()->get_height()),
+                )
+            );
+        }
+        else {
+            $self->main_window->resize($width, $height);
+        }
+    } else {
+        $self->debug("run_current_step: starting go_button_pressed...");
+        $self->current_step->go_button_pressed;
+        $self->debug("run_current_step: go_button_pressed exited.");
     }
 }
 
-sub goto_next_step {
-    my $self = shift;
-    (my $opts = shift) ||= {};
-
+method goto_next_step () {
     my $next_step;
 
     if ($next_step = $self->shift_steps) {
@@ -664,6 +677,13 @@ sub goto_next_step {
     }
     else {
         $self->debug("No more steps.");
+        if (! $self->display_finished_message) {
+            if ($self->gui) {
+                Gtk3->main_quit;
+            } else {
+                return;
+            }
+        }
         $self->current_step->title->set_text($self->encoding->decode(gettext(
             q{Persistence wizard - Finished}
         )));
@@ -678,10 +698,7 @@ You may now close this application.}
     }
 }
 
-sub step_object_from_name {
-    my $self = shift;
-    my $name = shift;
-
+method step_object_from_name (Str $name) {
     my $class_name = step_name_to_class_name($name);
 
     my %init_args;
@@ -692,10 +709,17 @@ sub step_object_from_name {
                 $self->create_persistence_partition({ @_ })
             },
             size_of_free_space          => $self->size_of_free_space,
-            should_mount_persistence_partition =>
-                0 < $self->grep_steps(sub { $_ eq 'configure' }),
             mount_persistence_partition_cb => sub {
-                $self->mount_persistence_partition({ @_ })
+                $self->mount_persistence_partition
+            },
+            create_configuration_cb        => sub {
+                $self->save_configuration;
+                my $asp_config_file = path(
+                    $self->persistence_partition_mountpoint,
+                    'live-additional-software.conf'
+                );
+                $asp_config_file->touch;
+                $asp_config_file->chmod(0644);
             },
         );
     }
@@ -711,9 +735,7 @@ sub step_object_from_name {
     }
     elsif ($name eq 'configure') {
         %init_args = (
-            go_callback => sub {
-                $self->save_configuration({ @_ })
-            },
+            go_callback                => sub { $self->save_configuration },
             configuration              => $self->configuration,
             persistence_partition      => $self->persistence_partition,
             persistence_partition_device_file => $self->persistence_partition_device_file,
@@ -724,7 +746,7 @@ sub step_object_from_name {
     return $class_name->new(
         name             => $name,
         encoding         => $self->encoding,
-        success_callback => sub { $self->goto_next_step({ @_ }) },
+        success_callback => sub { $self->goto_next_step },
         drive_vendor     => $self->boot_drive_vendor,
         drive_model      => $self->boot_drive_model,
         %init_args
@@ -733,7 +755,7 @@ sub step_object_from_name {
 }
 
 method get_variable_from_persistence_state_file (Str $variable) {
-    get_variable_from_file($self->persistence_state_file, $variable);
+    get_variable_from_file($self->persistence_state_file->stringify, $variable);
 }
 
 method persistence_filesystem_is_mounted () {
@@ -766,5 +788,5 @@ method persistence_filesystem_is_writable () {
     return $ret;
 }
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Step/Bootstrap.pm b/usr/share/perl5/Tails/Persistence/Step/Bootstrap.pm
index 7a456ee..c4139b3 100644
--- a/usr/share/perl5/Tails/Persistence/Step/Bootstrap.pm
+++ b/usr/share/perl5/Tails/Persistence/Step/Bootstrap.pm
@@ -5,51 +5,87 @@ Tails::Persistence::Step::Bootstrap - bootstrap persistent storage
 =cut
 
 package Tails::Persistence::Step::Bootstrap;
-use Moose;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
-
 
+use 5.10.1;
+use strictures 2;
 
+use Function::Parameters;
 use Glib qw{TRUE FALSE};
 
 use IPC::System::Simple qw{systemx};
 use Number::Format qw(:subs);
+use Types::Standard qw(HashRef);
 
 use Locale::gettext;
 use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
+use Moo;
+use MooX::late;
+use namespace::clean;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
 foreach (qw{intro label verify_label warning_label}) {
-    has $_ => lazy_build rw 'Gtk3::Label';
+    has $_ => (
+        lazy_build => 1,
+        is         => 'rw',
+        isa        => 'Gtk3::Label',
+    );
 }
 
 foreach (qw{passphrase_entry verify_passphrase_entry}) {
-    has $_ => lazy_build rw 'Gtk3::Entry', builder '_build_passphrase_entry';
+    has $_ => (
+        lazy_build => 1,
+        is         => 'rw',
+        isa        => 'Gtk3::Entry',
+        builder    => '_build_passphrase_entry',
+    );
 }
 
-has 'table_alignment' => lazy_build rw 'Gtk3::Alignment';
-has 'table'           => lazy_build rw 'Gtk3::Table';
-has 'warning_area'    => lazy_build rw 'Gtk3::HBox';
-has 'warning_image'   => lazy_build rw 'Gtk3::Image';
-has 'size_of_free_space'      => required ro Int;
-has 'mount_persistence_partition_cb' => required ro CodeRef;
-has 'should_mount_persistence_partition' => required ro Bool;
+has 'table_alignment' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::Alignment',
+);
+has 'table' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::Table',
+);
+has 'warning_area' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::HBox',
+);
+has 'warning_image' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::Image',
+);
+has 'size_of_free_space' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Int',
+);
+foreach (qw{mount_persistence_partition_cb create_configuration_cb}) {
+    has $_ => (
+        required => 1,
+        is       => 'ro',
+        isa      => 'CodeRef',
+    );
+}
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-sub BUILD {
-    my $self = shift;
-
+method BUILD (@args) {
     $self->title->set_text($self->encoding->decode(gettext(
         q{Persistence wizard - Persistent volume creation}
     )));
@@ -66,10 +102,7 @@ sub BUILD {
     $self->go_button->set_label($self->encoding->decode(gettext(q{Create})));
 }
 
-sub _build_main_box {
-    my $self = shift;
-
-
+method _build_main_widget () {
     my $box = Gtk3::VBox->new();
     $box->set_spacing(6);
     $box->pack_start($self->title, FALSE, FALSE, 0);
@@ -97,9 +130,7 @@ sub _build_main_box {
     return $box;
 }
 
-sub _build_intro {
-    my $self = shift;
-
+method _build_intro () {
     my $intro = Gtk3::Label->new('');
     $intro->set_alignment(0.0, 0.5);
     $intro->set_padding(10, 10);
@@ -107,21 +138,17 @@ sub _build_intro {
     $intro->set_line_wrap_mode('word');
     $intro->set_single_line_mode(FALSE);
     $intro->set_markup($self->encoding->decode(gettext(
-        q{<b>Beware!</b> Using persistence has consequences that must be well understood. Tails can't help you if you use it wrong! See <a href='file:///usr/share/doc/tails/website/doc/first_steps/persistence.en.html'>Tails documentation about persistence</a> to learn more.}
+        q{<b>Beware!</b> Using persistence has consequences that must be well understood. Tails can't help you if you use it wrong! See the <i>Encrypted persistence</i> page of the Tails documentation to learn more.}
     )));
 
     return $intro;
 }
 
-sub _build_warning_image {
-    my $self = shift;
-
+method _build_warning_image () {
     Gtk3::Image->new_from_stock('gtk-dialog-info', 'menu');
 }
 
-sub _build_warning_area {
-    my $self = shift;
-
+method _build_warning_area () {
     my $ext_hbox = Gtk3::HBox->new(FALSE, 0);
     $ext_hbox->set_border_width(10);
     $ext_hbox->set_spacing(6);
@@ -137,9 +164,7 @@ sub _build_warning_area {
     return $ext_hbox;
 }
 
-sub _build_label {
-    my $self = shift;
-
+method _build_label () {
     my $label = Gtk3::Label->new($self->encoding->decode(gettext(
         q{Passphrase:}
     )));
@@ -147,9 +172,7 @@ sub _build_label {
     return $label;
 }
 
-sub _build_verify_label {
-    my $self = shift;
-
+method _build_verify_label () {
     my $label = Gtk3::Label->new($self->encoding->decode(gettext(
         q{Verify Passphrase:}
     )));
@@ -157,9 +180,7 @@ sub _build_verify_label {
     return $label;
 }
 
-sub _build_warning_label {
-    my $self = shift;
-
+method _build_warning_label () {
     my $label = Gtk3::Label->new('');
     $label->set_padding(10, 0);
     $label->set_markup(
@@ -170,9 +191,7 @@ sub _build_warning_label {
     return $label;
 }
 
-sub _build_passphrase_entry {
-    my $self = shift;
-
+method _build_passphrase_entry () {
     my $entry = Gtk3::Entry->new;
     $entry->set_visibility(FALSE);
     $entry->signal_connect("changed" => sub { $self->passphrase_entry_changed });
@@ -180,17 +199,13 @@ sub _build_passphrase_entry {
     return $entry;
 }
 
-sub _build_table_alignment {
-    my $self = shift;
-
+method _build_table_alignment () {
     my $table_alignment = Gtk3::Alignment->new(0.0, 0.0, 1.0, 1.0);
     $table_alignment->add($self->table);
     return $table_alignment;
 }
 
-sub _build_table {
-    my $self = shift;
-
+method _build_table () {
     my $table = Gtk3::Table->new(1, 2, FALSE);
     $table->set_border_width(10);
     $table->set_col_spacings(12);
@@ -208,9 +223,7 @@ sub _build_table {
 
 =cut
 
-sub update_passphrase_ui {
-    my $self = shift;
-
+method update_passphrase_ui () {
     my $passphrase = $self->passphrase_entry->get_text;
     my $passphrase_verify = $self->verify_passphrase_entry->get_text;
 
@@ -241,15 +254,11 @@ sub update_passphrase_ui {
     }
 }
 
-sub passphrase_entry_changed {
-    my $self = shift;
-
+method passphrase_entry_changed () {
     $self->update_passphrase_ui;
 }
 
-sub operation_finished {
-    my $self    = shift;
-    my $replies = shift;
+method operation_finished (HashRef $replies) {
     my ($created_device, $error);
 
     say STDERR "Entering Bootstrap::operation_finished";
@@ -278,42 +287,45 @@ sub operation_finished {
         say STDERR "created ${created_device}.";
         $self->working(0);
 
-        if ($self->should_mount_persistence_partition) {
-            $self->subtitle->set_text($self->encoding->decode(gettext(
-                q{Mounting Tails persistence partition.}
-            )));
-            $self->description->set_text($self->encoding->decode(gettext(
-                q{The Tails persistence partition will be mounted.}
-            )));
-            $self->working(1);
-            systemx(qw{/sbin/udevadm settle});
-            my $mountpoint = $self->mount_persistence_partition_cb->();
-            $self->working(0);
-            say STDERR "mounted persistence partition on $mountpoint";
-
-            $self->subtitle->set_text($self->encoding->decode(gettext(
-                q{Correcting permissions of the persistent volume.}
-            )));
-            $self->description->set_text($self->encoding->decode(gettext(
-                q{The permissions of the persistent volume will be corrected.}
-            )));
-            $self->working(1);
-            systemx(qw{sudo -n /usr/bin/tails-fix-persistent-volume-permissions});
-            $self->working(0);
-            say STDERR "fixed permissions.";
-
-        }
-        else {
-            say STDERR "Should lock Luks device.";
-        }
+        $self->subtitle->set_text($self->encoding->decode(gettext(
+            q{Mounting Tails persistence partition.}
+        )));
+        $self->description->set_text($self->encoding->decode(gettext(
+            q{The Tails persistence partition will be mounted.}
+        )));
+        $self->working(1);
+        systemx(qw{/sbin/udevadm settle});
+        my $mountpoint = $self->mount_persistence_partition_cb->();
+        $self->working(0);
+        say STDERR "mounted persistence partition on $mountpoint";
+
+        $self->subtitle->set_text($self->encoding->decode(gettext(
+            q{Correcting permissions of the persistent volume.}
+        )));
+        $self->description->set_text($self->encoding->decode(gettext(
+            q{The permissions of the persistent volume will be corrected.}
+        )));
+        $self->working(1);
+        systemx(qw{sudo -n /usr/bin/tails-fix-persistent-volume-permissions});
+        $self->working(0);
+        say STDERR "fixed permissions.";
+
+        $self->subtitle->set_text($self->encoding->decode(gettext(
+            q{Creating default persistence configuration.}
+        )));
+        $self->description->set_text($self->encoding->decode(gettext(
+            q{The default persistence configuration will be created.}
+        )));
+        $self->working(1);
+        $self->create_configuration_cb->();
+        $self->working(0);
+        say STDERR "created default persistence configuration.";
 
         $self->success_callback->();
     }
 }
 
-sub go_button_pressed {
-    my $self = shift;
-
+method go_button_pressed () {
     $_->hide foreach ($self->intro, $self->warning_area, $self->table);
     $self->working(1);
     $self->subtitle->set_text(
@@ -332,5 +344,5 @@ sub go_button_pressed {
 with 'Tails::Persistence::Role::StatusArea';
 with 'Tails::Persistence::Role::SetupStep';
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Step/Configure.pm b/usr/share/perl5/Tails/Persistence/Step/Configure.pm
index 751d8e3..4b74948 100644
--- a/usr/share/perl5/Tails/Persistence/Step/Configure.pm
+++ b/usr/share/perl5/Tails/Persistence/Step/Configure.pm
@@ -5,18 +5,16 @@ Tails::Persistence::Step::Configure - configure which bits are persistent
 =cut
 
 package Tails::Persistence::Step::Configure;
-use Moose;
-use MooseX::Method::Signatures;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
 
-
-
-use Glib qw{TRUE FALSE};
+use 5.10.1;
+use strictures 2;
 
 use Carp::Assert::More;
+use Function::Parameters;
+use Glib qw{TRUE FALSE};
 use Number::Format qw(:subs);
-use Tails::Persistence::Configuration::Button;
+use Tails::Persistence::Configuration;
+use Tails::Persistence::Configuration::Setting;
 use Try::Tiny;
 
 use Locale::gettext;
@@ -24,38 +22,67 @@ use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
+use Moo;
+use MooX::late;
+use MooX::HandlesVia;
+use namespace::clean;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'configuration' => required ro 'Tails::Persistence::Configuration';
-
-has 'persistence_partition_device_file' => required ro Str;
-has 'persistence_partition_size' => required ro Int;
-
-has 'list_box' => lazy_build ro 'Gtk3::VBox';
-
-has 'buttons'  =>
-    lazy_build ro 'ArrayRef[Tails::Persistence::Configuration::Button]',
-    traits  => [ 'Array' ],
-    handles => {
-        all_buttons => 'elements',
-        push_button => 'push',
-    };
+has 'configuration' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Tails::Persistence::Configuration',
+);
+
+has 'persistence_partition_device_file' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Str',
+);
+has 'persistence_partition_size' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Int',
+);
+
+has 'list_box' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::ListBox',
+);
+
+has 'settings_container' => (
+    lazy_build => 1,
+    is         => 'ro',
+    isa        => 'Gtk3::ScrolledWindow',
+);
+
+has 'settings' => (
+    lazy_build  => 1,
+    is          => 'ro',
+    isa         => 'ArrayRef[Tails::Persistence::Configuration::Setting]',
+    handles_via => [ 'Array' ],
+    handles     => {
+        all_settings  => 'elements',
+        push_settings => 'push',
+    },
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-sub BUILD {
-    my $self = shift;
-
+method BUILD (@args) {
     # Force initialization in the right order
     assert_defined($self->configuration);
-    assert_defined($self->buttons);
+    assert_defined($self->settings);
     assert_defined($self->list_box);
+    assert_defined($self->settings_container);
 
     $self->title->set_text($self->encoding->decode(gettext(
         q{Persistence wizard - Persistent volume configuration}
@@ -75,21 +102,26 @@ sub BUILD {
     $self->go_button->set_sensitive(TRUE);
 }
 
-method _build_main_box {
-    my $box = Gtk3::VBox->new();
-    $box->set_spacing(6);
-    $box->pack_start($self->title, FALSE, FALSE, 0);
-    $box->pack_start($self->subtitle, FALSE, FALSE, 0);
-    $box->pack_start($self->description, FALSE, FALSE, 0);
-
+method _build_settings_container () {
     my $viewport = Gtk3::Viewport->new;
     $viewport->set_shadow_type('GTK_SHADOW_NONE');
     $viewport->add($self->list_box);
+
     my $scrolled_win = Gtk3::ScrolledWindow->new;
     $scrolled_win->set_policy('automatic', 'automatic');
+    $scrolled_win->set_overlay_scrolling(FALSE);
     $scrolled_win->add($viewport);
-    $box->pack_start($scrolled_win, TRUE, TRUE, 0);
 
+    return $scrolled_win;
+}
+
+method _build_main_widget () {
+    my $box = Gtk3::VBox->new();
+    $box->set_spacing(6);
+    $box->pack_start($self->title, FALSE, FALSE, 0);
+    $box->pack_start($self->subtitle, FALSE, FALSE, 0);
+    $box->pack_start($self->description, FALSE, FALSE, 0);
+    $box->pack_start($self->settings_container, TRUE, TRUE, 0);
     $box->pack_start($self->status_area, FALSE, FALSE, 0);
 
     my $button_alignment = Gtk3::Alignment->new(1.0, 0, 0.2, 1.0);
@@ -100,21 +132,41 @@ method _build_main_box {
     return $box;
 }
 
-method _build_buttons {
-    [
-        map {
-            Tails::Persistence::Configuration::Button->new(atom => $_)
-        } $self->configuration->all_atoms
-    ]
+method _build_settings () {
+    my @settings;
+    foreach my $preset ($self->configuration->presets->all) {
+        my %init_args;
+        if (exists($preset->{configuration_app_desktop_id})
+                && defined($preset->{configuration_app_desktop_id})) {
+            $init_args{configuration_app_desktop_id} = $preset->{configuration_app_desktop_id};
+        }
+        push @settings,
+            Tails::Persistence::Configuration::Setting->new(
+                id          => $preset->{id},
+                atoms       => $preset->{atoms},
+                name        => $preset->{name},
+                description => $preset->{description},
+                icon_name   => $preset->{icon_name},
+                enabled     => $preset->{enabled},
+                %init_args,
+            );
+    }
+    foreach my $atom ($self->configuration->atoms_not_in_presets) {
+        push @settings,
+            Tails::Persistence::Configuration::Setting->new(atoms => [$atom]);
+    }
+    return \@settings;
 }
 
-method _build_list_box {
-    my $box = Gtk3::VBox->new();
-    $box->set_spacing(6);
-    $box->pack_start($_->main_widget, FALSE, FALSE, 0) foreach $self->all_buttons;
-    $self->refresh_buttons;
+method _build_list_box () {
+    my $list_box = Gtk3::ListBox->new();
+    $list_box->set_selection_mode('GTK_SELECTION_NONE');
+    foreach my $setting ($self->all_settings) {
+        $list_box->insert($setting->main_widget, -1);
+        $list_box->insert(Gtk3::Separator->new('GTK_ORIENTATION_HORIZONTAL'), -1);
+    }
 
-    return $box;
+    return $list_box;
 }
 
 
@@ -122,7 +174,7 @@ method _build_list_box {
 
 =cut
 
-method operation_finished ($error) {
+method operation_finished ($error = undef) {
     if ($error) {
         $self->working(0);
         say STDERR "$error";
@@ -136,8 +188,8 @@ method operation_finished ($error) {
     }
 }
 
-method go_button_pressed {
-    $self->list_box->hide;
+method go_button_pressed () {
+    $self->settings_container->hide;
     $self->working(1);
     $self->subtitle->set_text(
         $self->encoding->decode(gettext(q{Saving...})),
@@ -155,12 +207,8 @@ method go_button_pressed {
     $self->operation_finished($error);
 }
 
-method refresh_buttons {
-    $_->refresh_display foreach $self->all_buttons;
-}
-
 with 'Tails::Persistence::Role::StatusArea';
 with 'Tails::Persistence::Role::SetupStep';
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Step/Delete.pm b/usr/share/perl5/Tails/Persistence/Step/Delete.pm
index 88916ff..8a6511e 100644
--- a/usr/share/perl5/Tails/Persistence/Step/Delete.pm
+++ b/usr/share/perl5/Tails/Persistence/Step/Delete.pm
@@ -5,14 +5,12 @@ Tails::Persistence::Step::Delete - delete persistent storage
 =cut
 
 package Tails::Persistence::Step::Delete;
-use Moose;
-use MooseX::Types::Moose qw( :all );
-use MooseX::Has::Sugar::Saccharin;
-
 
+use 5.10.1;
+use strictures 2;
 
+use Function::Parameters;
 use Glib qw{TRUE FALSE};
-
 use Number::Format qw(:subs);
 
 use Locale::gettext;
@@ -20,23 +18,37 @@ use POSIX;
 setlocale(LC_MESSAGES, "");
 textdomain("tails-persistence-setup");
 
+use Moo;
+use MooX::late;
+use namespace::clean;
+
 
 =head1 ATTRIBUTES
 
 =cut
 
-has 'persistence_partition_device_file' => required ro Str;
-has 'persistence_partition_size' => required ro Int;
-has 'warning_icon'               => lazy_build rw 'Gtk3::Image';
+has 'persistence_partition_device_file' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Str',
+);
+has 'persistence_partition_size' => (
+    required => 1,
+    is       => 'ro',
+    isa      => 'Int',
+);
+has 'warning_icon' => (
+    lazy_build => 1,
+    is         => 'rw',
+    isa        => 'Gtk3::Image',
+);
 
 
 =head1 CONSTRUCTORS
 
 =cut
 
-sub BUILD {
-    my $self = shift;
-
+method BUILD (@args) {
     $self->title->set_text($self->encoding->decode(gettext(
         q{Persistence wizard - Persistent volume deletion}
     )));
@@ -55,13 +67,11 @@ sub BUILD {
     $self->go_button->set_sensitive(TRUE);
 }
 
-sub _build_warning_icon {
+method _build_warning_icon () {
     Gtk3::Image->new_from_stock("gtk-dialog-warning", "GTK_ICON_SIZE_DIALOG");
 }
 
-sub _build_main_box {
-    my $self = shift;
-
+method _build_main_widget () {
     my $box = Gtk3::VBox->new();
     my $hbox = Gtk3::HBox->new();
     $box->set_spacing(6);
@@ -82,9 +92,7 @@ sub _build_main_box {
     return $box;
 }
 
-sub operation_finished {
-    my $self  = shift;
-    my $reply = shift;
+method operation_finished ($reply) {
     my $error;
 
     { local $@; eval { $reply->get_result }; $error = $@; }
@@ -103,9 +111,7 @@ sub operation_finished {
     }
 }
 
-sub go_button_pressed {
-    my $self = shift;
-
+method go_button_pressed () {
     $self->working(1);
     $self->subtitle->set_text(
         $self->encoding->decode(gettext(q{Deleting...})),
@@ -122,5 +128,5 @@ sub go_button_pressed {
 with 'Tails::Persistence::Role::StatusArea';
 with 'Tails::Persistence::Role::SetupStep';
 
-no Moose;
+no Moo;
 1;
diff --git a/usr/share/perl5/Tails/Persistence/Utils.pm b/usr/share/perl5/Tails/Persistence/Utils.pm
index a58b02e..9e48c43 100644
--- a/usr/share/perl5/Tails/Persistence/Utils.pm
+++ b/usr/share/perl5/Tails/Persistence/Utils.pm
@@ -6,65 +6,52 @@ Tails::Persistence::Utils - utilities for Tails persistent storage
 
 package Tails::Persistence::Utils;
 
-use strict;
-use warnings FATAL => 'all';
+use strictures 2;
 use 5.10.1;
 
+use Carp;
+use Function::Parameters;
+use IPC::System::Simple qw{capturex};
+use List::MoreUtils qw{last_value};
+use Path::Tiny;
+use Types::Standard qw(HashRef Int Str StrictNum);
+use Types::Path::Tiny qw{Path};
 
+use namespace::clean;
 
 use Exporter;
 our @ISA = qw{Exporter};
 our @EXPORT = qw{align_up_at_2MiB align_down_at_2MiB step_name_to_class_name get_variable_from_file check_config_file_permissions};
 
-use Carp;
-use IPC::System::Simple qw{capturex};
-use List::MoreUtils qw{last_value};
-use Path::Class;
-
 
 =head1 FUNCTIONS
 
 =cut
 
-sub round_down {
-    my $number = shift;
-    my $round = shift;
-
+fun round_down (StrictNum $number, StrictNum $round) {
     return (int($number/$round)) * $round if $number % $round;
     return $number;
 }
 
-sub round_up {
-    my $number = shift;
-    my $round = shift;
-
+fun round_up (StrictNum $number, StrictNum $round) {
     return (1 + int($number/$round)) * $round if $number % $round;
     return $number;
 }
 
-sub align_up_at_2MiB {
-    my $bytes = shift;
-
+fun align_up_at_2MiB (Int $bytes) {
     round_up($bytes, 2 * 2 ** 20)
 }
 
-sub align_down_at_2MiB {
-    my $bytes = shift;
-
+fun align_down_at_2MiB (Int $bytes) {
     round_down($bytes, 2 * 2 ** 20)
 }
 
-sub step_name_to_class_name {
-    my $name = shift;
-
+fun step_name_to_class_name (Str $name) {
     'Tails::Persistence::Step::' . ucfirst($name);
 }
 
-sub get_variable_from_file {
-    my $file     = shift;
-    my $variable = shift;
-
-    foreach my $line (file($file)->slurp(chomp => 1)) {
+fun get_variable_from_file (Str $file, Str $variable) {
+    foreach my $line (path($file)->lines({chomp => 1})) {
         if (my ($name, $value) =
                 ($line =~ m{\A [[:space:]]* ($variable)=(.*) \z}xms)) {
             return $value;
@@ -81,11 +68,7 @@ or ownership than expected. We don't check the parent directory since
 live-persist ensures it's right for us.
 
 =cut
-sub check_config_file_permissions {
-    @_ == 2 or croak(q{check_config_file_permissions needs 2 arguments});
-    my $config_file_path = shift;
-    my $expected         = shift;
-
+fun check_config_file_permissions (Path $config_file_path, HashRef $expected) {
     croak(q{persistence.conf is a symbolic link})  if -l $config_file_path;
     croak(q{persistence.conf is not a plain file}) unless -f $config_file_path;
 
@@ -94,6 +77,8 @@ sub check_config_file_permissions {
     # ownership
     foreach my $field (qw{uid gid}) {
         my $actual_value = $st->$field;
+        croak("Expected value for '$field' is not defined")
+            unless defined $expected->{$field};
         $actual_value eq $expected->{$field}
             or croak("persistence.conf has unsafe $field: '$actual_value'");
     }
