--- /etc/xdg/menus/gnome-applications.menu 2019-08-26 23:17:27.532000000 +0000
+++ /etc/xdg/menus/gnome-applications.menu	2019-08-26 23:18:41.156000000 +0000
@@ -35,7 +35,6 @@
            Please keep the list alphabetically sorted! -->
       <Filename>eog.desktop</Filename>
       <Filename>gucharmap.desktop</Filename>
-      <Filename>org.gnome.Calculator.desktop</Filename>
       <Filename>org.gnome.DejaDup.desktop</Filename>
       <Filename>org.gnome.Dictionary.desktop</Filename>
       <Filename>org.gnome.DiskUtility.desktop</Filename>
@@ -295,7 +294,6 @@
       <Filename>gnome-system-log.desktop</Filename>
       <Filename>gnome-system-monitor.desktop</Filename>
       <Filename>org.gnome.baobab.desktop</Filename>
-      <Filename>org.gnome.Terminal.desktop</Filename>
     </Exclude>

     <Menu>
@@ -414,7 +412,6 @@
       <Filename>gnome-system-monitor.desktop</Filename>
       <Filename>gucharmap.desktop</Filename>
       <Filename>org.gnome.baobab.desktop</Filename>
-      <Filename>org.gnome.Calculator.desktop</Filename>
       <Filename>org.gnome.DejaDup.desktop</Filename>
       <Filename>org.gnome.Dictionary.desktop</Filename>
       <Filename>org.gnome.DiskUtility.desktop</Filename>
@@ -422,7 +419,6 @@
       <Filename>org.gnome.FileRoller.desktop</Filename>
       <Filename>org.gnome.fonts.desktop</Filename>
       <Filename>org.gnome.Screenshot.desktop</Filename>
-      <Filename>org.gnome.Terminal.desktop</Filename>
       <Filename>org.gnome.tweaks.desktop</Filename>
       <Filename>seahorse.desktop</Filename>
       <Filename>vinagre.desktop</Filename>
