--- chroot.orig/etc/init.d/alsa-utils	2010-10-04 18:14:10.000000000 +0000
+++ chroot/etc/init.d/alsa-utils	2012-03-12 13:49:11.815997232 +0000
@@ -206,5 +206,10 @@
 	# On MacBookPro5,3 and later models (See Bug#597791)
 	unmute_and_set_level "Front Speaker" "80%"
+       # On MacBook5,2 models (See Bug#602973)
+       unmute_and_set_level "LFE" "80%"
 
+	# On Intel 82801H (See Bug#603550)
+	unmute_and_set_level "Speaker" "80%"
+
 	return 0
 }
