--- a/usr/share/applications/pidgin.desktop	2014-11-14 11:07:14.000000000 +0000
+++ b/usr/share/applications/pidgin.desktop	2014-11-26 17:13:35.000000000 +0000
@@ -183,7 +183,7 @@
 Comment[zh_CN]=互联网通讯程序。 支持 AIM、Google Talk、Jabber/XMPP、MSN、Yahoo 和更多
 Comment[zh_HK]=讓你可以透過即時通訊與好友聊天，支援 AIM、Google Talk、Jabber/XMPP、MSN、Yahoo 等等
 Comment[zh_TW]=讓您可以透過即時通訊與好友聊天，支援 AIM、Google Talk、Jabber/XMPP、MSN、Yahoo 等等
-Exec=pidgin
+Exec=/usr/local/bin/pidgin
 Icon=pidgin
 StartupNotify=true
 Terminal=false
