--- chroot.orig/usr/share/themes/XPLuna/gtk-2.0/panel.rc	2011-12-09 13:22:07.262967642 +0000
+++ chroot/usr/share/themes/XPLuna/gtk-2.0/panel.rc	2011-12-09 13:18:42.919889100 +0000
@@ -213,3 +213,32 @@

 widget_class "*Panel*GtkToggleButton" style "panelbuttons"
 widget "*.tasklist-button" style "panelbuttons"
+
+#############################################################
+#REMOVE UGLY HANDLES FROM THE PANEL
+#############################################################
+
+style "handle"
+{
+engine "pixmap"
+  {
+    image
+    {
+      function				= HANDLE
+      file              		= ""
+      border            		= { 0, 0, 0, 0 }
+      stretch           		= TRUE
+      orientation			= VERTICAL
+    }
+    image
+    {
+      function				= HANDLE
+      file              		= ""
+      border            		= { 0, 0, 0, 0 }
+      stretch           		= TRUE
+      orientation			= HORIZONTAL
+    }
+  }
+}
+
+class "PanelAppletFrame" style "handle"
