--- a/usr/share/zenity/zenity.ui
+++ b/usr/share/zenity/zenity.ui
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <interface>
-  <!-- interface-requires gtk+ 2.6 -->
+  <!-- interface-requires gtk+ 3.0 -->
   <object class="GtkAdjustment" id="adjustment1">
     <property name="upper">100</property>
     <property name="step_increment">1</property>
@@ -27,6 +27,7 @@
             <child>
               <object class="GtkButton" id="zenity_calendar_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -43,6 +44,7 @@
             <child>
               <object class="GtkButton" id="zenity_calendar_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="has_focus">True</property>
@@ -161,6 +163,7 @@
             <child>
               <object class="GtkButton" id="zenity_entry_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -177,6 +180,7 @@
             <child>
               <object class="GtkButton" id="zenity_entry_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -268,6 +272,7 @@
             <child>
               <object class="GtkButton" id="zenity_error_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -323,6 +328,7 @@
                     <property name="use_markup">True</property>
                     <property name="wrap">True</property>
                     <property name="selectable">True</property>
+                    <property name="ellipsize">start</property>
                   </object>
                   <packing>
                     <property name="expand">False</property>
@@ -367,6 +373,7 @@
             <child>
               <object class="GtkButton" id="zenity_forms_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">True</property>
@@ -382,6 +389,7 @@
             <child>
               <object class="GtkButton" id="zenity_forms_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">True</property>
@@ -427,6 +435,12 @@
                     <child>
                       <placeholder/>
                     </child>
+                    <child>
+                      <placeholder/>
+                    </child>
+                    <child>
+                      <placeholder/>
+                    </child>
                   </object>
                 </child>
               </object>
@@ -473,6 +487,7 @@
             <child>
               <object class="GtkButton" id="zenity_info_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -523,6 +538,7 @@
                 <property name="use_markup">True</property>
                 <property name="wrap">True</property>
                 <property name="selectable">True</property>
+                <property name="ellipsize">start</property>
               </object>
               <packing>
                 <property name="expand">False</property>
@@ -563,6 +579,7 @@
             <child>
               <object class="GtkButton" id="zenity_progress_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -579,6 +596,7 @@
             <child>
               <object class="GtkButton" id="zenity_progress_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="sensitive">False</property>
                 <property name="can_focus">True</property>
@@ -703,6 +721,7 @@
                 <property name="use_markup">True</property>
                 <property name="wrap">True</property>
                 <property name="selectable">True</property>
+                <property name="ellipsize">start</property>
               </object>
               <packing>
                 <property name="expand">False</property>
@@ -741,6 +760,7 @@
             <child>
               <object class="GtkButton" id="zenity_scale_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -757,6 +777,7 @@
             <child>
               <object class="GtkButton" id="zenity_scale_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -849,6 +870,7 @@
             <child>
               <object class="GtkButton" id="zenity_text_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
@@ -864,6 +886,7 @@
             <child>
               <object class="GtkButton" id="zenity_text_close_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -918,6 +941,7 @@
             </child>
             <child>
               <object class="GtkCheckButton" id="zenity_text_checkbox">
+                <property name="use_action_appearance">False</property>
                 <property name="can_focus">True</property>
                 <property name="receives_default">False</property>
                 <property name="use_action_appearance">False</property>
@@ -965,6 +989,7 @@
             <child>
               <object class="GtkButton" id="zenity_tree_cancel_button">
                 <property name="label">gtk-cancel</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -981,6 +1006,7 @@
             <child>
               <object class="GtkButton" id="zenity_tree_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
@@ -1078,6 +1104,7 @@
             <child>
               <object class="GtkButton" id="zenity_warning_ok_button">
                 <property name="label">gtk-ok</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="has_focus">True</property>
@@ -1130,6 +1157,7 @@
                 <property name="use_markup">True</property>
                 <property name="wrap">True</property>
                 <property name="selectable">True</property>
+                <property name="ellipsize">start</property>
               </object>
               <packing>
                 <property name="expand">False</property>
