--- /usr/share/zenity/zenity.ui	2021-01-22 12:22:18.403962128 +0000
+++ /usr/share/zenity/zenity.ui	2021-01-22 12:22:51.647960610 +0000
@@ -841,7 +841,7 @@
             <child>
               <object class="GtkProgressBar" id="zenity_progress_bar">
                 <property name="visible">True</property>
-                <property name="can_focus">False</property>
+                <property name="can_focus">True</property>
                 <property name="pulse_step">0.10000000149</property>
               </object>
               <packing>
