diff --git a/usr/bin/sudo-tar b/usr/bin/sudo-tar
deleted file mode 100755
index ac997e9..0000000
--- a/usr/bin/sudo-tar
+++ /dev/null
@@ -1,4 +0,0 @@
-#!/bin/sh
-
-umask 0022
-sudo -n tar $@
diff --git a/usr/bin/tails-create-iuk b/usr/bin/tails-create-iuk
index d584242..d7a249d 100755
--- a/usr/bin/tails-create-iuk
+++ b/usr/bin/tails-create-iuk
@@ -10,8 +10,7 @@ Version
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 our $VERSION = '4.0.3';
diff --git a/usr/bin/tails-install-iuk b/usr/bin/tails-install-iuk
index ac9bc25..a3381c6 100755
--- a/usr/bin/tails-install-iuk
+++ b/usr/bin/tails-install-iuk
@@ -6,8 +6,7 @@ tails-install-iuk - install an Incremental Upgrade Kit
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 use FindBin;
diff --git a/usr/bin/tails-iuk-check-upgrade-description-file b/usr/bin/tails-iuk-check-upgrade-description-file
index ac15e3f..626ddb4 100755
--- a/usr/bin/tails-iuk-check-upgrade-description-file
+++ b/usr/bin/tails-iuk-check-upgrade-description-file
@@ -6,8 +6,7 @@ tails-iuk-check-upgrade-description-file - check correctness of upgrade-descript
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 use FindBin;
@@ -19,7 +18,7 @@ use Carp::Assert::More;
 use Data::Dumper::Concise;
 use File::Temp qw{tempdir};
 use IPC::Run qw{run};
-use Path::Class qw{file};
+use Path::Tiny;
 use Tails::IUK::UpgradeDescriptionFile;
 use Tails::IUK::Utils;
 
@@ -32,7 +31,7 @@ assert_exists(
 assert_nonblank(
     $ENV{TAILS_SIGNATURE_KEY}, q{TAILS_SIGNATURE_KEY variable is not empty}
 );
-my $trusted_gnupg_homedir = tempdir;
+my $trusted_gnupg_homedir = Path::Tiny->tempdir;
 run [qw{gpg --batch --quiet --export}, $ENV{TAILS_SIGNATURE_KEY}],
     '|', [qw{gpg --batch --quiet --homedir}, $trusted_gnupg_homedir, '--import'];
 
@@ -45,9 +44,9 @@ for my $input_filename (@input_filenames) {
     );
     assert_isa($upgrade_description, 'Tails::IUK::UpgradeDescriptionFile');
 
-    my $signature_file = file($input_filename . '.pgp');
+    my $signature_file = path($input_filename . '.pgp');
     assert(-e $signature_file, "Signature for '$input_filename' exists.");
-    my $description_txt = file($input_filename)->slurp;
+    my $description_txt = path($input_filename)->slurp;
     my $signature_txt   = $signature_file->slurp;
     if (! verify_signature($description_txt, $signature_txt, $trusted_gnupg_homedir)) {
         push @failed, "$signature_file";
diff --git a/usr/bin/tails-iuk-generate-upgrade-description-files b/usr/bin/tails-iuk-generate-upgrade-description-files
index 1d6ee31..02614e6 100755
--- a/usr/bin/tails-iuk-generate-upgrade-description-files
+++ b/usr/bin/tails-iuk-generate-upgrade-description-files
@@ -6,8 +6,7 @@ tails-iuk-generate-upgrade-description-file - create and update upgrade-descript
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 use FindBin;
diff --git a/usr/bin/tails-iuk-get-target-file b/usr/bin/tails-iuk-get-target-file
index e747541..acd004f 100755
--- a/usr/bin/tails-iuk-get-target-file
+++ b/usr/bin/tails-iuk-get-target-file
@@ -6,8 +6,7 @@ tails-iuk-get-target-file - download and verify a target file
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 use FindBin;
diff --git a/usr/bin/tails-iuk-get-upgrade-description-file b/usr/bin/tails-iuk-get-upgrade-description-file
index 9963274..8c23741 100755
--- a/usr/bin/tails-iuk-get-upgrade-description-file
+++ b/usr/bin/tails-iuk-get-upgrade-description-file
@@ -6,8 +6,7 @@ tails-iuk-get-upgrade-description-file - download, verify and output an upgrade
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 use FindBin;
diff --git a/usr/bin/tails-upgrade-frontend b/usr/bin/tails-upgrade-frontend
index 04875fb..de64b9e 100755
--- a/usr/bin/tails-upgrade-frontend
+++ b/usr/bin/tails-upgrade-frontend
@@ -6,8 +6,7 @@ tails-upgrade-frontend - lead Tails user through the process of upgrading the sy
 
 =cut
 
-use strict;
-use warnings;
+use strictures 2;
 use 5.10.1;
 
 use FindBin;
